<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Genotype-Phenotype Integration for Antimicrobial Resistance Research • AMRgen</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="deps/Nunito-0.4.10/font.css" rel="stylesheet">
<link href="deps/Fira_Code-0.4.10/font.css" rel="stylesheet">
<link href="deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="deps/headroom-0.11.0/headroom.min.js"></script><script src="deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="deps/search-1.0.0/fuse.min.js"></script><script src="deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Genotype-Phenotype Integration for Antimicrobial Resistance Research">
<meta name="description" content="The AMRgen package provides tools for interpreting antimicrobial resistance (AMR) genes, integrating genomic data with phenotypic antimicrobial susceptibility testing (AST) data, and calculating genotype-phenotype associations. This package extends the functionality of the AMR package, enabling researchers to fetch MIC distribution data from EUCAST, visualise multidrug resistance patterns using UpSet plots, and generate NCBI-compliant antibiograms. Designed for AMR research and surveillance, AMRgen aims to support data-driven decision-making in combating AMR on a global scale.">
<meta property="og:description" content="The AMRgen package provides tools for interpreting antimicrobial resistance (AMR) genes, integrating genomic data with phenotypic antimicrobial susceptibility testing (AST) data, and calculating genotype-phenotype associations. This package extends the functionality of the AMR package, enabling researchers to fetch MIC distribution data from EUCAST, visualise multidrug resistance patterns using UpSet plots, and generate NCBI-compliant antibiograms. Designed for AMR research and surveillance, AMRgen aims to support data-driven decision-making in combating AMR on a global scale.">
<meta property="og:image" content="https://AMRverse.github.io/AMRgen/logo.png">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="index.html">AMRgen</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.1.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="index.html"><span class="fa fa-home"></span> Home</a></li>
<li class="nav-item"><a class="nav-link" href="reference/index.html"><span class="fa fa-book-open"></span> Reference</a></li>
<li class="nav-item"><a class="nav-link" href="articles/AnalysingGenoPhenoData.html"><span class="fa fa-book"></span> Vignettes</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/AMRverse/AMRgen"><span class="fa fab fa-github"></span> Source Code</a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header">
<img src="logo.png" class="logo" alt=""><h1 id="amrgen">AMRgen<a class="anchor" aria-label="anchor" href="#amrgen"></a>
</h1>
</div>
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p><img src="logo.png" width="200" align="right" alt="AMRgen"></p>
<p><strong>AMRgen</strong> is an open-source R package designed to <strong>bridge the gap between genotypic and phenotypic antimicrobial resistance (AMR) data</strong>. Developed as an extension to the <a href="https://amr-for-r.org" class="external-link">AMR R package</a>, it provides tools to interpret AMR genes, integrate these findings with antimicrobial susceptibility test (AST) data, and calculate genotype-phenotype associations.</p>
<p>This package is developed in collaboration with the ESGEM-AMR Working Group and is tailored for researchers and healthcare professionals tackling AMR globally.</p>
<p>The <a href="https://AMRverse.github.io/AMRgen">AMRgen website</a> has full function <a href="https://AMRverse.github.io/AMRgen/reference/index.html">documentation</a> and a <a href="https://AMRverse.github.io/AMRgen/articles/AnalysingGenoPhenoData.html">Vignette</a> working through analysing geno/pheno data using key functions.</p>
<hr>
</div>
<div class="section level2">
<h2 id="key-features">Key Features<a class="anchor" aria-label="anchor" href="#key-features"></a>
</h2>
<ul>
<li>
<strong>Import Genotype and Phenotype Data</strong>: Import from common formats (NCBI or EBI antibiogram format for phenotypes; AMRFinderPlus and hAMRonization for genotypes</li>
<li>
<strong>Genotype-Phenotype Integration</strong>: Links AMR gene presence with phenotypic resistance profiles, enabling deeper insights into resistance mechanisms.</li>
<li>
<strong>Automated EUCAST MIC Distribution Integration</strong>: Fetch MIC distribution data directly from <a href="https://mic.eucast.org" class="external-link">EUCAST</a> for seamless comparison with local susceptibility data.</li>
<li>
<strong>Visualisation</strong>: Generate powerful UpSet plots to identify intersections of AMR gene presence and phenotypic resistance, highlighting multidrug resistance patterns.</li>
<li>
<strong>Modular and Extensible</strong>: Leverages the robust foundation of the AMR package, including antibiotic selectors and clinical breakpoint interpretations.</li>
</ul>
<p>Planned for development:</p>
<ul>
<li>
<strong>NCBI-Compliant Export</strong>: Export phenotype data to NCBI-compliant antibiogram format.</li>
<li>
<strong>Expanded Data Import</strong>: Import and parse phenotype data from other tools (e.g. CARD, ResFinder).</li>
</ul>
<hr>
</div>
<div class="section level2">
<h2 id="getting-started">Getting Started<a class="anchor" aria-label="anchor" href="#getting-started"></a>
</h2>
<p>To install and explore the package, follow the instructions below:</p>
<div class="section level3">
<h3 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h3>
<p>It is best to restart R before running the installation to prevent issues.</p>
<p>Install the latest version of this package with:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"remotes"</span><span class="op">)</span> <span class="co"># if you haven't already</span></span>
<span><span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"AMRverse/AMRgen"</span><span class="op">)</span></span></code></pre></div>
<p>All required packages, including the <a href="https://amr-for-r.org" class="external-link">AMR package</a> if you don’t have it already, will be installed automatically.</p>
<p>If you have issues, we recommend you install the latest version of the AMR package directly, then try again to install AMRgen:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"remotes"</span><span class="op">)</span> <span class="co"># if you haven't already</span></span>
<span><span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"msberends/AMR"</span><span class="op">)</span></span>
<span><span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"AMRverse/AMRgen"</span><span class="op">)</span></span></code></pre></div>
<p>If you still have trouble with installation please post an issue <a href="https://github.com/AMRverse/AMRgen/issues" class="external-link">here</a>! The package is new and we want to make it as accessible as possible for new users.</p>
</div>
</div>
<div class="section level2">
<h2 id="quick-usage-examples">Quick Usage Examples<a class="anchor" aria-label="anchor" href="#quick-usage-examples"></a>
</h2>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://AMRverse.github.io/AMRgen/">AMRgen</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="investigate-ciprofloxacin-resistance-vs-quinolone-genotype-markers-via-solo-ppv-and-upset-plots">Investigate ciprofloxacin resistance vs quinolone genotype markers, via solo PPV and upset plots<a class="anchor" aria-label="anchor" href="#investigate-ciprofloxacin-resistance-vs-quinolone-genotype-markers-via-solo-ppv-and-upset-plots"></a>
</h3>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Example public E. coli AST data from NCBI</span></span>
<span><span class="co">#  (already imported via import_ncbi_ast() and re-interpreted with as.sir())</span></span>
<span><span class="va">ecoli_ast</span></span>
<span></span>
<span><span class="co"># Import matching E. coli AMRFinderPlus data from AllTheBacteria</span></span>
<span><span class="va">ecoli_geno</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/import_amrfp.html">import_amrfp</a></span><span class="op">(</span><span class="va">ecoli_geno_raw</span>, <span class="st">"Name"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Calculate solo positive predictive value for ciprofloxacin resistance, for individual markers found solo</span></span>
<span><span class="co">#  (for all quinolone-associated genotype markers)</span></span>
<span><span class="va">soloPPV_cipro</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/solo_ppv_analysis.html">solo_ppv_analysis</a></span><span class="op">(</span><span class="va">ecoli_geno</span>, <span class="va">ecoli_ast</span>, antibiotic<span class="op">=</span><span class="st">"Ciprofloxacin"</span>, drug_class_list<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Quinolones"</span><span class="op">)</span>, sir_col<span class="op">=</span><span class="st">"pheno_clsi"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Do upset plot of ciprofloxacin MIC vs quinolone genotype marker combinations</span></span>
<span><span class="co">#  (for combinations observed at least 5 times)</span></span>
<span><span class="va">cip_upset</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/amr_upset.html">amr_upset</a></span><span class="op">(</span><span class="va">soloPPV_cipro</span><span class="op">$</span><span class="va">amr_binary</span>, min_set_size<span class="op">=</span><span class="fl">5</span>, assay<span class="op">=</span><span class="st">"mic"</span>, order<span class="op">=</span><span class="st">"value"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Calculate positive predictive value for individual markers and combinations</span></span>
<span><span class="va">cip_ppv</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/ppv.html">ppv</a></span><span class="op">(</span><span class="va">soloPPV_cipro</span><span class="op">$</span><span class="va">amr_binary</span>, min_set_size<span class="op">=</span><span class="fl">5</span>, assay<span class="op">=</span><span class="st">"mic"</span>, order<span class="op">=</span><span class="st">"value"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Do logistic regression of ciprofloxacin resistance as a function of presence/absence of quinolone-associated markers</span></span>
<span><span class="co">#  (for markers observed at least 10 times)</span></span>
<span><span class="va">models</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/amr_logistic.html">amr_logistic</a></span><span class="op">(</span>geno_table <span class="op">=</span> <span class="fu"><a href="reference/import_amrfp.html">import_amrfp</a></span><span class="op">(</span><span class="va">ecoli_geno_raw</span>, <span class="st">"Name"</span><span class="op">)</span>,</span>
<span>                       pheno_table <span class="op">=</span> <span class="va">ecoli_ast</span>, sir_col<span class="op">=</span><span class="st">"pheno_clsi"</span>,</span>
<span>                       antibiotic <span class="op">=</span> <span class="st">"Ciprofloxacin"</span>, drug_class_list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Quinolones"</span><span class="op">)</span>, maf<span class="op">=</span><span class="fl">10</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="download-reference-mic-distribution-from-eucastorg-and-compare-to-example-data">Download reference MIC distribution from eucast.org and compare to example data<a class="anchor" aria-label="anchor" href="#download-reference-mic-distribution-from-eucastorg-and-compare-to-example-data"></a>
</h3>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get MIC reference distribution for ciprofloxacin in E. coli</span></span>
<span><span class="va">ecoli_cip_mic_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/get_eucast_amr_distribution.html">get_eucast_mic_distribution</a></span><span class="op">(</span><span class="st">"cipro"</span>, <span class="st">"E. coli"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot the reference distribution </span></span>
<span><span class="va">mics</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="va">ecoli_cip_mic_data</span><span class="op">$</span><span class="va">mic</span>, <span class="va">ecoli_cip_mic_data</span><span class="op">$</span><span class="va">count</span><span class="op">)</span></span>
<span><span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/autoplot.html" class="external-link">autoplot</a></span><span class="op">(</span><span class="va">mics</span>, ab <span class="op">=</span> <span class="st">"cipro"</span>, mo <span class="op">=</span> <span class="st">"E. coli"</span>, title <span class="op">=</span> <span class="st">"E. coli cipro reference distribution"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Compare reference distribution to example E. coli data</span></span>
<span><span class="va">ecoli_cip</span> <span class="op">&lt;-</span> <span class="va">ecoli_ast</span><span class="op">$</span><span class="va">mic</span><span class="op">[</span><span class="va">ecoli_ast</span><span class="op">$</span><span class="va">drug_agent</span><span class="op">==</span><span class="st">"CIP"</span><span class="op">]</span></span>
<span><span class="va">comparison</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/get_eucast_amr_distribution.html">compare_mic_with_eucast</a></span><span class="op">(</span><span class="va">ecoli_cip</span>, ab <span class="op">=</span> <span class="st">"cipro"</span>, mo <span class="op">=</span> <span class="st">"E. coli"</span><span class="op">)</span></span>
<span><span class="va">comparison</span></span>
<span><span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/autoplot.html" class="external-link">autoplot</a></span><span class="op">(</span><span class="va">comparison</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="download-phenotype-or-genotype-data-from-ebis-amr-portal">Download phenotype or genotype data from EBI’s AMR portal<a class="anchor" aria-label="anchor" href="#download-phenotype-or-genotype-data-from-ebis-amr-portal"></a>
</h3>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get phenotype data for ciprofloxacin in E. coli</span></span>
<span><span class="va">ebi_pheno_ecoli_cip</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/download_ebi.html">download_ebi</a></span><span class="op">(</span>species <span class="op">=</span> <span class="st">"Escherichia coli"</span>, antibiotic<span class="op">=</span><span class="st">"Cipro"</span>, reformat<span class="op">=</span><span class="cn">T</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Check how many observations we have for MIC data</span></span>
<span><span class="va">ebi_pheno_ecoli_cip</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">mic</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot MIC distribution, stratified by platform</span></span>
<span><span class="fu"><a href="reference/assay_by_var.html">assay_by_var</a></span><span class="op">(</span><span class="va">ebi_pheno_ecoli_cip</span>, measure<span class="op">=</span><span class="st">"mic"</span>, colour_by <span class="op">=</span> <span class="st">"pheno_provided"</span>, facet_var <span class="op">=</span> <span class="st">"platform"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Compare MIC values with reference distribution from EUCAST</span></span>
<span><span class="va">compare_mics</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/get_eucast_amr_distribution.html">compare_mic_with_eucast</a></span><span class="op">(</span><span class="va">ebi_pheno_ecoli_cip</span><span class="op">$</span><span class="va">mic</span>, ab <span class="op">=</span> <span class="st">"cipro"</span>, mo <span class="op">=</span> <span class="st">"E. coli"</span><span class="op">)</span></span>
<span><span class="va">compare_mics</span></span>
<span><span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/autoplot.html" class="external-link">autoplot</a></span><span class="op">(</span><span class="va">compare_mics</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Compare disk diffusion zone values with reference distribution from EUCAST</span></span>
<span><span class="va">compare_disk</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/get_eucast_amr_distribution.html">compare_disk_with_eucast</a></span><span class="op">(</span><span class="va">ebi_pheno_ecoli_cip</span><span class="op">$</span><span class="va">disk</span>, ab <span class="op">=</span> <span class="st">"cipro"</span>, mo <span class="op">=</span> <span class="st">"E. coli"</span><span class="op">)</span></span>
<span><span class="va">compare_disk</span></span>
<span><span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/autoplot.html" class="external-link">autoplot</a></span><span class="op">(</span><span class="va">compare_disk</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get genotype data for ciprofloxacin in E. coli</span></span>
<span><span class="va">ebi_geno</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/download_ebi.html">download_ebi</a></span><span class="op">(</span>data<span class="op">=</span><span class="st">"genotype"</span>, species <span class="op">=</span> <span class="st">"Escherichia coli"</span>, geno_subclass<span class="op">=</span><span class="st">"QUINOLONE"</span>, reformat<span class="op">=</span><span class="cn">T</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create binary matrix summarising cipro geno and pheno data from EBI</span></span>
<span><span class="va">cipro_ebi_geno_pheno</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/get_binary_matrix.html">get_binary_matrix</a></span><span class="op">(</span><span class="va">ecoli_geno</span>, <span class="va">ebi_pheno_ecoli_cip</span>, antibiotic<span class="op">=</span><span class="st">"Ciprofloxacin"</span>, drug_class_list<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Quinolones"</span><span class="op">)</span>, sir_col<span class="op">=</span><span class="st">"pheno_provided"</span>, keep_assay_values<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Do upset plot of ciprofloxacin MIC vs quinolone genotype marker combinations</span></span>
<span><span class="co">#  (for combinations observed at least 5 times)</span></span>
<span><span class="va">cip_upset_ebi_mic</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/amr_upset.html">amr_upset</a></span><span class="op">(</span><span class="va">cipro_ebi_geno_pheno</span>, min_set_size<span class="op">=</span><span class="fl">5</span>, assay<span class="op">=</span><span class="st">"mic"</span>, order<span class="op">=</span><span class="st">"value"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="import-and-export">Import and export<a class="anchor" aria-label="anchor" href="#import-and-export"></a>
</h3>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Import phenotype data in NCBI, EBI, Vitek, WHOnet formats</span></span>
<span><span class="op">?</span><span class="va">import_ast</span></span>
<span></span>
<span><span class="co"># Export phenotype data in NCBI or EBI formats</span></span>
<span><span class="op">?</span><span class="va">export_ast</span></span></code></pre></div>
<p>For more see the <a href="https://AMRverse.github.io/AMRgen/articles/AnalysingGenoPhenoData.html">Vignette</a>.</p>
</div>
</div>
<div class="section level2">
<h2 id="contributions">Contributions<a class="anchor" aria-label="anchor" href="#contributions"></a>
</h2>
<p>Contributions are welcome! If you encounter issues or wish to suggest new features, please open an issue or submit a pull request.</p>
</div>
<div class="section level2">
<h2 id="licence">Licence<a class="anchor" aria-label="anchor" href="#licence"></a>
</h2>
<p>This package is distributed under the GNU GPL-3.0 Licence. See <code>LICENSE</code> for details.</p>
</div>
</div>
  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/AMRverse/AMRgen/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/AMRverse/AMRgen/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li>
<a href="https://www.r-project.org/Licenses/GPL-3" class="external-link">GPL-3</a> | file <a href="LICENSE-text.html">LICENSE</a>
</li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing AMRgen</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Matthijs S. Berends <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0001-7620-1800" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a>  </li>
<li>Natacha Couto <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0002-9152-5464" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a>  </li>
<li>Zoe A. Dyson <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0002-8887-3492" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a>  </li>
<li>Jane Hawkey <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0001-9661-5293" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a>  </li>
<li>Kat Holt <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0003-3949-2471" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a>  </li>
</ul>
</div>



  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p>Free and Open Source (GPL-3), brought by AMRverse/AMRgen Development Team</p>
</div>

<div class="pkgdown-footer-right">
  <p>Our goal is to develop interpretive standards for AMR genotypes, akin to the interpretive standards developed by EUCAST and CLSI for antimicrobial susceptibility phenotyping.</p>
</div>

    </footer>
</div>





  </body>
</html>
