<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>AMRgen: Genotype-Phenotype Integration for Antimicrobial Resistance Research — AMRgen-package • AMRgen</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/Nunito-0.4.10/font.css" rel="stylesheet"><link href="../deps/Fira_Code-0.4.10/font.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="AMRgen: Genotype-Phenotype Integration for Antimicrobial Resistance Research — AMRgen-package"><meta name="description" content="The AMRgen package provides tools for interpreting antimicrobial resistance (AMR) genes, integrating genomic data with phenotypic antimicrobial susceptibility testing (AST) data, and calculating genotype-phenotype associations. This package extends the functionality of the AMR package, enabling researchers to fetch MIC distribution data from EUCAST, visualise multidrug resistance patterns using UpSet plots, and generate NCBI-compliant antibiograms. Designed for AMR research and surveillance, AMRgen aims to support data-driven decision-making in combating AMR on a global scale."><meta property="og:description" content="The AMRgen package provides tools for interpreting antimicrobial resistance (AMR) genes, integrating genomic data with phenotypic antimicrobial susceptibility testing (AST) data, and calculating genotype-phenotype associations. This package extends the functionality of the AMR package, enabling researchers to fetch MIC distribution data from EUCAST, visualise multidrug resistance patterns using UpSet plots, and generate NCBI-compliant antibiograms. Designed for AMR research and surveillance, AMRgen aims to support data-driven decision-making in combating AMR on a global scale."><meta property="og:image" content="https://AMRverse.github.io/AMRgen/logo.png"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">AMRgen</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.1.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../index.html"><span class="fa fa-home"></span> Home</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html"><span class="fa fa-book-open"></span> Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../articles/AnalysingGenoPhenoData.html"><span class="fa fa-book"></span> Vignettes</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><a class="external-link nav-link" href="https://github.com/AMRverse/AMRgen"><span class="fa fab fa-github"></span> Source Code</a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>AMRgen: Genotype-Phenotype Integration for Antimicrobial Resistance Research</h1>
      <small class="dont-index">Source: <a href="https://github.com/AMRverse/AMRgen/blob/main/R/AMRgen-package.R" class="external-link"><code>R/AMRgen-package.R</code></a></small>
      <div class="d-none name"><code>AMRgen-package.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>The AMRgen package provides tools for interpreting antimicrobial resistance (AMR) genes, integrating genomic data with phenotypic antimicrobial susceptibility testing (AST) data, and calculating genotype-phenotype associations. This package extends the functionality of the AMR package, enabling researchers to fetch MIC distribution data from EUCAST, visualise multidrug resistance patterns using UpSet plots, and generate NCBI-compliant antibiograms. Designed for AMR research and surveillance, AMRgen aims to support data-driven decision-making in combating AMR on a global scale.</p>
    </div>


    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p>Useful links:</p><ul><li><p><a href="https://AMRverse.github.io/AMRgen/">https://AMRverse.github.io/AMRgen/</a></p></li>
<li><p><a href="https://github.com/AMRverse/AMRgen" class="external-link">https://github.com/AMRverse/AMRgen</a></p></li>
<li><p>Report bugs at <a href="https://github.com/AMRverse/AMRgen/issues" class="external-link">https://github.com/AMRverse/AMRgen/issues</a></p></li>
</ul></div>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p><strong>Maintainer</strong>: Matthijs S. Berends <a href="mailto:m.s.berends@umcg.nl">m.s.berends@umcg.nl</a> (<a href="https://orcid.org/0000-0001-7620-1800" class="external-link">ORCID</a>)</p>
<p>Authors:</p><ul><li><p>Natacha Couto (<a href="https://orcid.org/0000-0002-9152-5464" class="external-link">ORCID</a>)</p></li>
<li><p>Zoe A. Dyson <a href="mailto:zoe.dyson@lshtm.ac.uk">zoe.dyson@lshtm.ac.uk</a> (<a href="https://orcid.org/0000-0002-8887-3492" class="external-link">ORCID</a>)</p></li>
<li><p>Jane Hawkey (<a href="https://orcid.org/0000-0001-9661-5293" class="external-link">ORCID</a>)</p></li>
<li><p>Kat Holt (<a href="https://orcid.org/0000-0003-3949-2471" class="external-link">ORCID</a>)</p></li>
</ul></div>

  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Free and Open Source (GPL-3), brought by AMRverse/AMRgen Development Team</p>
</div>

<div class="pkgdown-footer-right">
  <p>Our goal is to develop interpretive standards for AMR genotypes, akin to the interpretive standards developed by EUCAST and CLSI for antimicrobial susceptibility phenotyping.</p>
</div>

    </footer></div>





  </body></html>

