<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Import EBI-processed AMRFinderPlus Genotypes from Web — import_amrfp_ebi_web • AMRgen</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/Nunito-0.4.10/font.css" rel="stylesheet"><link href="../deps/Fira_Code-0.4.10/font.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Import EBI-processed AMRFinderPlus Genotypes from Web — import_amrfp_ebi_web"><meta name="description" content="This function imports EBI-processed AMRFinderPlus genotyping results. The expected input is genotype data downloaded from the EBI AMR Portal web browser.
Note that files downloaded from the EBI AMR Portal FTP site, either directly or via the function download_ebi(), are formatted differently and can be imported using import_amrfp_ebi_ftp."><meta property="og:description" content="This function imports EBI-processed AMRFinderPlus genotyping results. The expected input is genotype data downloaded from the EBI AMR Portal web browser.
Note that files downloaded from the EBI AMR Portal FTP site, either directly or via the function download_ebi(), are formatted differently and can be imported using import_amrfp_ebi_ftp."><meta property="og:image" content="https://AMRverse.github.io/AMRgen/logo.png"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">AMRgen</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.1.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../index.html"><span class="fa fa-home"></span> Home</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html"><span class="fa fa-book-open"></span> Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../articles/AnalysingGenoPhenoData.html"><span class="fa fa-book"></span> Vignettes</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><a class="external-link nav-link" href="https://github.com/AMRverse/AMRgen"><span class="fa fab fa-github"></span> Source Code</a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Import EBI-processed AMRFinderPlus Genotypes from Web</h1>
      <small class="dont-index">Source: <a href="https://github.com/AMRverse/AMRgen/blob/main/R/import_amrfp.R" class="external-link"><code>R/import_amrfp.R</code></a></small>
      <div class="d-none name"><code>import_amrfp_ebi_web.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function imports EBI-processed AMRFinderPlus genotyping results. The expected input is genotype data downloaded from the <a href="https://www.ebi.ac.uk/amr/data/?view=predictions" class="external-link">EBI AMR Portal web browser</a>.
Note that files downloaded from the <a href="https://ftp.ebi.ac.uk/pub/databases/amr_portal/releases/" class="external-link">EBI AMR Portal FTP site</a>, either directly or via the function <code><a href="download_ebi.html">download_ebi()</a></code>, are formatted differently and can be imported using <a href="import_amrfp_ebi_ftp.html">import_amrfp_ebi_ftp</a>.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">import_amrfp_ebi_web</span><span class="op">(</span><span class="va">input_table</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-input-table">input_table<a class="anchor" aria-label="anchor" href="#arg-input-table"></a></dt>
<dd><p>R object or file path for the input EBI genotype table (R object, or file path to a TSV or CSV file).</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A tibble containing the processed AMR elements, with harmonised gene names, drug agents, and drug classes. The output retains the original columns from the AMRFinderPlus table along with the newly mapped variables.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>These data are pre-processed by EBI to match NCBI class/subclass to CARD's antibiotic resistance ontology (ARO), however for consistency this function will re-process the data to generate <code>drug_agent</code> and <code>drug_class</code> fields consistent with the <code><a href="import_amrfp.html">import_amrfp()</a></code> function (the EBI fields <code>antibiotic*</code> are also retained).
Note several AMRfinderplus fields are excluded from EBI files, including hierarchy node, method, percent identity and coverage; therefore unlike the <code><a href="import_amrfp.html">import_amrfp()</a></code> function, this function cannot assign <code>variation type</code> or <code>node</code>.</p>
<p>The function performs the following steps:</p><ul><li><p>Reads the EBI-processed genotype table.</p></li>
<li><p>Maps AMRFinderPlus subclasses to standardised drug agent and drug class
names using <code>amrfp_drugs</code> (EBI-mappings are retained in <code>antibiotic*</code> fields.)</p></li>
<li><p>Converts drug agent names to the <code>ab</code> class from the AMR package.
This processing ensures compatibility with downstream AMR analysis workflows.</p></li>
</ul></div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span> <span class="co"># \dontrun{</span></span></span>
<span class="r-in"><span><span class="co"># Download data from EBI web portal and import the file</span></span></span>
<span class="r-in"><span><span class="va">ebi_geno</span> <span class="op">&lt;-</span> <span class="fu">import_amrfp_ebi_web</span><span class="op">(</span><span class="st">"amr_records.csv"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span> <span class="co"># }</span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Free and Open Source (GPL-3), brought by AMRverse/AMRgen Development Team</p>
</div>

<div class="pkgdown-footer-right">
  <p>Our goal is to develop interpretive standards for AMR genotypes, akin to the interpretive standards developed by EUCAST and CLSI for antimicrobial susceptibility phenotyping.</p>
</div>

    </footer></div>





  </body></html>

