<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Package index â€¢ AMRgen</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/Nunito-0.4.10/font.css" rel="stylesheet"><link href="../deps/Fira_Code-0.4.10/font.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Package index"><meta property="og:image" content="https://AMRverse.github.io/AMRgen/logo.png"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">AMRgen</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.1.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../index.html"><span class="fa fa-home"></span> Home</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html"><span class="fa fa-book-open"></span> Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../articles/AnalysingGenoPhenoData.html"><span class="fa fa-book"></span> Vignettes</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><a class="external-link nav-link" href="https://github.com/AMRverse/AMRgen"><span class="fa fab fa-github"></span> Source Code</a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-index">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Package index</h1>
    </div>

    <div class="section level2">
      <h2 id="data-import-preprocessing">Data Import &amp; Preprocessing<a class="anchor" aria-label="anchor" href="#data-import-preprocessing"></a></h2>

      <div class="section-desc"><p>Functions to import, harmonise, and prepare genotype and phenotype data from public repositories or internal formats.</p></div>

      <dl></dl></div><div class="section level2">




      <dl><dt>

          <code><a href="import_amrfp.html">import_amrfp()</a></code>

        </dt>
        <dd>Import and Process AMRFinderPlus Results</dd>

        <dt>

          <code><a href="import_amrfp_ebi.html">import_amrfp_ebi()</a></code>

        </dt>
        <dd>Import EBI-processed AMRFinderPlus Genotypes</dd>

        <dt>

          <code><a href="import_amrfp_ebi_ftp.html">import_amrfp_ebi_ftp()</a></code>

        </dt>
        <dd>Import EBI-processed AMRFinderPlus Genotypes from FTP</dd>

        <dt>

          <code><a href="import_amrfp_ebi_web.html">import_amrfp_ebi_web()</a></code>

        </dt>
        <dd>Import EBI-processed AMRFinderPlus Genotypes from Web</dd>

        <dt>

          <code><a href="import_ast.html">import_ast()</a></code>

        </dt>
        <dd>Import and Process AST Data from an EBI or NCBI antibiogram File</dd>

        <dt>

          <code><a href="import_ebi_ast.html">import_ebi_ast()</a></code>

        </dt>
        <dd>Import and Process AST Data from files downloaded from the EBI AMR portal website</dd>

        <dt>

          <code><a href="import_ebi_ast_ftp.html">import_ebi_ast_ftp()</a></code>

        </dt>
        <dd>Import and Process AST Data files retrieved from the EBI AMR portal FTP site</dd>

        <dt>

          <code><a href="import_gtdb.html">import_gtdb()</a></code>

        </dt>
        <dd>Import GTDB Output</dd>

        <dt>

          <code><a href="import_ncbi_ast.html">import_ncbi_ast()</a></code>

        </dt>
        <dd>Import and Process AST Data from an NCBI File</dd>

        <dt>

          <code><a href="import_vitek_ast.html">import_vitek_ast()</a></code>

        </dt>
        <dd>Import and Process AST Data from Vitek Output Files</dd>

        <dt>

          <code><a href="import_whonet_ast.html">import_whonet_ast()</a></code>

        </dt>
        <dd>Import and Process AST Data from WHONET Output Files</dd>

        <dt>

          <code><a href="export_ast.html">export_ast()</a></code>

        </dt>
        <dd>Export AST Data</dd>

        <dt>

          <code><a href="export_ebi_antibiogram.html">export_ebi_antibiogram()</a></code>

        </dt>
        <dd>Export EBI Antibiogram</dd>

        <dt>

          <code><a href="export_ncbi_biosample.html">export_ncbi_biosample()</a></code>

        </dt>
        <dd>Export NCBI BioSample Antibiogram</dd>

        <dt>

          <code><a href="convert_aa_code.html">convert_aa_code()</a></code>

        </dt>
        <dd>Convert single-letter amino acid code(s) to three-letter code(s)</dd>

        <dt>

          <code><a href="convert_mutation.html">convert_mutation()</a></code>

        </dt>
        <dd>Convert mutation string based on method</dd>

        <dt>

          <code><a href="get_binary_matrix.html">get_binary_matrix()</a></code>

        </dt>
        <dd>Get Binary Matrix of Genotype and Phenotype Data</dd>

        <dt>

          <code><a href="get_combo_matrix.html">get_combo_matrix()</a></code>

        </dt>
        <dd>Add marker combinations to a binary geno-pheno matrix</dd>

        <dt>

          <code><a href="combo_stats.html">combo_stats()</a></code>

        </dt>
        <dd>Generate a Series of Plots for AMR Gene and Combination Analysis</dd>

        <dt>

          <code><a href="as.gene.html">as.gene()</a></code>

        </dt>
        <dd>Gene Class and AMR Parsing Functions</dd>

        <dt>

          <code><a href="gtdb.mo.html">gtdb.mo()</a></code>

        </dt>
        <dd>Get Microorganism from GTDB Species Name</dd>

        <dt>

          <code><a href="format_ast.html">format_ast()</a></code>

        </dt>
        <dd>Import and Process AST Data from a generic format</dd>

        <dt>

          <code><a href="download_ebi.html">download_ebi()</a></code>

        </dt>
        <dd>Download antimicrobial genotype or phenotype data from the EBI AMR Portal</dd>
      </dl></div><div class="section level2">
      <h2 id="resistance-interpretation">Resistance Interpretation<a class="anchor" aria-label="anchor" href="#resistance-interpretation"></a></h2>

      <div class="section-desc"><p>Core tools for interpreting antimicrobial resistance based on EUCAST breakpoints and custom models.</p></div>

      <dl></dl></div><div class="section level2">




      <dl><dt>

          <code><a href="compare_estimates.html">compare_estimates()</a></code>

        </dt>
        <dd>Plot to Compare Two Sets of Estimates</dd>

        <dt>

          <code><a href="compare_geno_pheno_id.html">compare_geno_pheno_id()</a></code>

        </dt>
        <dd>Compare Genotype and Phenotype Data by Sample ID</dd>

        <dt>

          <code><a href="get_eucast_amr_distribution.html">get_eucast_amr_distribution()</a></code> <code><a href="get_eucast_amr_distribution.html">get_eucast_mic_distribution()</a></code> <code><a href="get_eucast_amr_distribution.html">get_eucast_disk_distribution()</a></code> <code><a href="get_eucast_amr_distribution.html">compare_mic_with_eucast()</a></code> <code><a href="get_eucast_amr_distribution.html">compare_disk_with_eucast()</a></code>

        </dt>
        <dd>Get and Compare Antimicrobial Wild Type Distributions from EUCAST</dd>

        <dt>

          <code><a href="eucast_supported_ab_distributions.html">eucast_supported_ab_distributions()</a></code>

        </dt>
        <dd>Retrieve Available Antimicrobial Wild Type Distributions from EUCAST</dd>

        <dt>

          <code><a href="merge_logreg_soloppv.html">merge_logreg_soloppv()</a></code>

        </dt>
        <dd>Merge Logistic Regression and Solo PPV Statistics</dd>

        <dt>

          <code><a href="interpret_ast.html">interpret_ast()</a></code>

        </dt>
        <dd>Interpret AST data in a standard format tibble</dd>
      </dl></div><div class="section level2">
      <h2 id="modelling-and-analysis">Modelling and analysis<a class="anchor" aria-label="anchor" href="#modelling-and-analysis"></a></h2>

      <div class="section-desc"><p>Statistical models for resistance prediction and inference, including logistic regression and Firth regression.</p></div>

      <dl></dl></div><div class="section level2">




      <dl><dt>

          <code><a href="amr_upset.html">amr_upset()</a></code>

        </dt>
        <dd>Generate Upset Plot</dd>

        <dt>

          <code><a href="solo_ppv_analysis.html">solo_ppv_analysis()</a></code>

        </dt>
        <dd>Perform Solo PPV Analysis for AMR Markers</dd>

        <dt>

          <code><a href="ppv.html">ppv()</a></code>

        </dt>
        <dd>Generate Upset Plot</dd>

        <dt>

          <code><a href="amr_logistic.html">amr_logistic()</a></code>

        </dt>
        <dd>AMR Logistic Regression Analysis</dd>

        <dt>

          <code><a href="glm_details.html">glm_details()</a></code>

        </dt>
        <dd>Extract Details from a Generalized Linear Model</dd>

        <dt>

          <code><a href="logistf_details.html">logistf_details()</a></code>

        </dt>
        <dd>Extract Details from a logistf Model</dd>

        <dt>

          <code><a href="getBreakpoints.html">getBreakpoints()</a></code>

        </dt>
        <dd>Get Clinical Breakpoints for an Antibiotic</dd>

        <dt>

          <code><a href="checkBreakpoints.html">checkBreakpoints()</a></code>

        </dt>
        <dd>Check and Retrieve Breakpoints for an Antibiotic</dd>
      </dl></div><div class="section level2">
      <h2 id="visualisation-reporting">Visualisation &amp; Reporting<a class="anchor" aria-label="anchor" href="#visualisation-reporting"></a></h2>

      <div class="section-desc"><p>Tools to visualise results, including MIC distributions, model outputs, and genotype-phenotype relationships.</p></div>

      <dl></dl></div><div class="section level2">




      <dl><dt>

          <code><a href="plot_combined_stats.html">plot_combined_stats()</a></code>

        </dt>
        <dd>Plot Combined Statistics</dd>

        <dt>

          <code><a href="plot_estimates.html">plot_estimates()</a></code>

        </dt>
        <dd>Plot Estimates from a Table of Results</dd>

        <dt>

          <code><a href="plot_solo_logReg.html">plot_solo_logReg()</a></code>

        </dt>
        <dd>Plot Combined Statistics of Logistic Regression and Solo PPV</dd>

        <dt>

          <code><a href="assay_by_var.html">assay_by_var()</a></code>

        </dt>
        <dd>Generate a Stacked Bar Plot of Assay Values Colored by a Variable</dd>
      </dl></div><div class="section level2">
      <h2 id="data">Data<a class="anchor" aria-label="anchor" href="#data"></a></h2>

      <div class="section-desc"><p>Example datasets for demonstration and reproducible analysis.</p></div>

      <dl></dl></div><div class="section level2">




      <dl><dt>

          <code><a href="ecoli_ast.html">ecoli_ast</a></code>

        </dt>
        <dd>E. coli NCBI AST Example Data, Re-interpreted with AMR Package</dd>

        <dt>

          <code><a href="ecoli_ast_raw.html">ecoli_ast_raw</a></code>

        </dt>
        <dd>E. coli NCBI AST Example Data</dd>

        <dt>

          <code><a href="ecoli_geno_raw.html">ecoli_geno_raw</a></code>

        </dt>
        <dd>E. coli Genotype Example Data</dd>
      </dl></div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Free and Open Source (GPL-3), brought by AMRverse/AMRgen Development Team</p>
</div>

<div class="pkgdown-footer-right">
  <p>Our goal is to develop interpretive standards for AMR genotypes, akin to the interpretive standards developed by EUCAST and CLSI for antimicrobial susceptibility phenotyping.</p>
</div>

    </footer></div>





  </body></html>

