<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Analysing Geno-Pheno Data • AMRgen</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/Nunito-0.4.10/font.css" rel="stylesheet">
<link href="../deps/Fira_Code-0.4.10/font.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Analysing Geno-Pheno Data">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">AMRgen</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.1.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../index.html"><span class="fa fa-home"></span> Home</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html"><span class="fa fa-book-open"></span> Reference</a></li>
<li class="active nav-item"><a class="nav-link" href="../articles/AnalysingGenoPhenoData.html"><span class="fa fa-book"></span> Vignettes</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/AMRverse/AMRgen"><span class="fa fab fa-github"></span> Source Code</a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Analysing Geno-Pheno Data</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/AMRverse/AMRgen/blob/main/vignettes/AnalysingGenoPhenoData.Rmd" class="external-link"><code>vignettes/AnalysingGenoPhenoData.Rmd</code></a></small>
      <div class="d-none name"><code>AnalysingGenoPhenoData.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p><code>AMRgen</code> is a comprehensive R package designed to
integrate antimicrobial resistance genotype and phenotype data. It
provides tools to:</p>
<ul>
<li><p>Import AMR genotype data (e.g. from AMRFinderPlus,
hAMRonization)</p></li>
<li><p>Import AST phenotype data (e.g. public data from NCBI or EBI, or
your own data in formats like Vitek or WHOnet)</p></li>
<li><p>Conduct genotype-phenotype analyses to explore the impact of
genotypic markers on phenotype, including via logistic regression, solo
marker analysis, and upset plots</p></li>
<li><p>Fetch MIC or disk zone reference distributions from
EUCAST</p></li>
</ul>
<p>This vignette walks through a basic workflow using example datasets
included in the <code>AMRgen</code> package, and explains how to wrangle
your own data files into the right formats to use the same workflow.</p>
<p>Start by loading the package:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://AMRverse.github.io/AMRgen/">AMRgen</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'dplyr'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:stats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     filter, lag</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     intersect, setdiff, setequal, union</span></span></code></pre></div>
<div class="section level3">
<h3 id="genotype-table">1. Genotype table<a class="anchor" aria-label="anchor" href="#genotype-table"></a>
</h3>
<p>The <code><a href="../reference/import_amrfp.html">import_amrfp()</a></code> function lets you load genotype data
from AMRFinderPlus output files, and process it to generate an object
with the key columns needed to work with the <code>AMRgen</code>
package.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Example AMRFinderPlus genotyping output (from Allthebacteria project)</span></span>
<span><span class="va">ecoli_geno_raw</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 45,228 × 28</span></span></span>
<span><span class="co">#&gt;    Name      `Protein identifier` `Contig id`  Start   Stop Strand `Gene symbol`</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>                <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> SAMN0317… <span style="color: #BB0000;">NA</span>                   SAMN031776…  <span style="text-decoration: underline;">74</span>721  <span style="text-decoration: underline;">75</span>851 -      blaEC        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> SAMN0317… <span style="color: #BB0000;">NA</span>                   SAMN031776… <span style="text-decoration: underline;">166</span>214 <span style="text-decoration: underline;">169</span>315 +      acrF         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> SAMN0317… <span style="color: #BB0000;">NA</span>                   SAMN031776…  <span style="text-decoration: underline;">20</span>678  <span style="text-decoration: underline;">22</span>033 -      glpT_E448K   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> SAMN0317… <span style="color: #BB0000;">NA</span>                   SAMN031776…    758   <span style="text-decoration: underline;">1</span>969 -      floR         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> SAMN0317… <span style="color: #BB0000;">NA</span>                   SAMN031776…   <span style="text-decoration: underline;">4</span>440   <span style="text-decoration: underline;">5</span>666 +      mdtM         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> SAMN0317… <span style="color: #BB0000;">NA</span>                   SAMN031776…   <span style="text-decoration: underline;">3</span>941   <span style="text-decoration: underline;">4</span>798 +      blaTEM-1     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> SAMN0317… <span style="color: #BB0000;">NA</span>                   SAMN031776…    142    954 +      sul2         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> SAMN0317… <span style="color: #BB0000;">NA</span>                   SAMN031776…   <span style="text-decoration: underline;">1</span>018   <span style="text-decoration: underline;">1</span>818 +      aph(3'')-Ib  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> SAMN0317… <span style="color: #BB0000;">NA</span>                   SAMN031776…   <span style="text-decoration: underline;">1</span>821   <span style="text-decoration: underline;">2</span>654 +      aph(6)-Id    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> SAMN0317… <span style="color: #BB0000;">NA</span>                   SAMN031776…    788   <span style="text-decoration: underline;">1</span>957 +      tet(A)       </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 45,218 more rows</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 21 more variables: `Sequence name` &lt;chr&gt;, Scope &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Element type` &lt;chr&gt;, `Element subtype` &lt;chr&gt;, Class &lt;chr&gt;, Subclass &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   Method &lt;chr&gt;, `Target length` &lt;dbl&gt;, `Reference sequence length` &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `% Coverage of reference sequence` &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `% Identity to reference sequence` &lt;dbl&gt;, `Alignment length` &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Accession of closest sequence` &lt;chr&gt;, `Name of closest sequence` &lt;chr&gt;, …</span></span></span>
<span></span>
<span><span class="co"># Load AMRFinderPlus output</span></span>
<span><span class="co">#    (replace 'ecoli_geno_raw' with the filepath for any AMRFinderPlus output)</span></span>
<span><span class="va">ecoli_geno</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/import_amrfp.html">import_amrfp</a></span><span class="op">(</span><span class="va">ecoli_geno_raw</span>, <span class="st">"Name"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Check the format of the processed genotype table</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">ecoli_geno</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 36</span></span></span>
<span><span class="co">#&gt;   Name         gene  mutation  node  marker   marker.label drug_agent drug_class</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;ab&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> SAMN03177615 blaEC <span style="color: #BB0000;">NA</span>        blaEC blaEC    blaEC        <span style="color: #BB0000;">NA</span>         Beta-lact…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> SAMN03177615 acrF  <span style="color: #BB0000;">NA</span>        acrF  acrF     acrF         <span style="color: #BB0000;">NA</span>         Efflux    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> SAMN03177615 glpT  Glu448Lys glpT  glpT_E4… glpT:Glu448… FOS        Other ant…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> SAMN03177615 floR  <span style="color: #BB0000;">NA</span>        floR  floR     floR         CHL        Amphenico…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> SAMN03177615 floR  <span style="color: #BB0000;">NA</span>        floR  floR     floR         FLR        Other ant…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> SAMN03177615 mdtM  <span style="color: #BB0000;">NA</span>        mdtM  mdtM     mdtM         <span style="color: #BB0000;">NA</span>         Efflux    </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 28 more variables: `Protein identifier` &lt;lgl&gt;, `Contig id` &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   Start &lt;dbl&gt;, Stop &lt;dbl&gt;, Strand &lt;chr&gt;, `Gene symbol` &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Sequence name` &lt;chr&gt;, Scope &lt;chr&gt;, `Element type` &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Element subtype` &lt;chr&gt;, Class &lt;chr&gt;, Subclass &lt;chr&gt;, Method &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `Target length` &lt;dbl&gt;, `Reference sequence length` &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `% Coverage of reference sequence` &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   `% Identity to reference sequence` &lt;dbl&gt;, `Alignment length` &lt;dbl&gt;, …</span></span></span></code></pre></div>
<p>The genotype table has one row for each genetic marker detected in an
input genome, i.e. one per strain/marker combination.</p>
<p>If your genotype data is not in AMRFinderPlus format, you can wrangle
other input data files into the necessary format.</p>
<p>The essential columns for a genotype table to work with
<code>AMRgen</code> functions are:</p>
<ul>
<li><p><code>Name</code>: character string giving the sample name, used
to link to sample names in the phenotype file (this column can have a
different name, in which case you’ll need to make sure it is the first
column in the dataframe OR pass its name to the functions using
<code>geno_sample_col</code>)</p></li>
<li><p><code>marker</code>: character string giving the name of the
genetic marker detected</p></li>
<li><p><code>drug_class</code>: character string giving the antibiotic
class associated with this marker</p></li>
</ul>
<p>NOTE: You should consider whether you have genomes with no AMR
markers detected by genotyping, and how to make sure these are include
in your analyses. E.g. AMRFinderPlus will output one row per
genome/marker combination, but if you have a genome with no markers
detected, there will be no row at all for that genome in the
concatenated output file. If your species has core genes included in
AMRFinderPlus this probably won’t be a problem as you would expect some
calls for every genome (e.g. AMRFinderPlus will report blaSHV, oqxA,
oqxB, fosA in all Klebsiella pneumoniae genomes, so all input genomes
will appear in the concatenated output file). An easy solution is to run
a check to make sure that all genome names in your input dataset are
represented in the genotype table, and if any are missing add empty rows
for these using
e.g. <code>tibble(Name=missing_samples) %&gt;% bind_rows(genotype_table)</code>.</p>
</div>
<div class="section level3">
<h3 id="phenotype-table">2. Phenotype table<a class="anchor" aria-label="anchor" href="#phenotype-table"></a>
</h3>
<p>The <code><a href="../reference/import_ncbi_ast.html">import_ncbi_ast()</a></code> function imports AST data from
NCBI format files.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Example E. coli AST data from NCBI</span></span>
<span><span class="co"># This one has already been imported and phenotypes interpreted from assay data</span></span>
<span><span class="co"># You can make your own from different file formats, and interpret against breakpoints, using:</span></span>
<span><span class="co">#    import_ast("filepath/NCBI_AST.tsv", format="ncbi", interpret_clsi=T)</span></span>
<span><span class="co">#    import_ast("filepath/Vitek_AST.tsv", format="vitek", interpret_eucast=T)</span></span>
<span></span>
<span><span class="va">ecoli_ast</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 4,170 × 10</span></span></span>
<span><span class="co">#&gt;    id           drug_agent     mic  disk pheno_clsi ecoff guideline method</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;ab&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;mic&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dsk&gt;</span> <span style="color: #949494; font-style: italic;">&lt;sir&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;sir&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> SAMN36015110 CIP        <span style="color: #555555;">&lt;</span>128<span style="color: #555555;">.00</span>    <span style="color: #BB0000;">NA</span> <span style="color: #080808; background-color: #FF5F5F;">  R  </span>      <span style="color: #080808; background-color: #FF5F5F;">  R  </span> CLSI      <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> SAMN11638310 CIP         256<span style="color: #555555;">.00</span>    <span style="color: #BB0000;">NA</span> <span style="color: #080808; background-color: #FF5F5F;">  R  </span>      <span style="color: #080808; background-color: #FF5F5F;">  R  </span> CLSI      <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> SAMN05729964 CIP          64<span style="color: #555555;">.00</span>    <span style="color: #BB0000;">NA</span> <span style="color: #080808; background-color: #FF5F5F;">  R  </span>      <span style="color: #080808; background-color: #FF5F5F;">  R  </span> CLSI      Etest </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> SAMN10620111 CIP         <span style="color: #555555;">&gt;=</span>4<span style="color: #555555;">.00</span>    <span style="color: #BB0000;">NA</span> <span style="color: #080808; background-color: #FF5F5F;">  R  </span>      <span style="color: #080808; background-color: #FF5F5F;">  R  </span> CLSI      <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> SAMN10620168 CIP         <span style="color: #555555;">&gt;=</span>4<span style="color: #555555;">.00</span>    <span style="color: #BB0000;">NA</span> <span style="color: #080808; background-color: #FF5F5F;">  R  </span>      <span style="color: #080808; background-color: #FF5F5F;">  R  </span> CLSI      <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> SAMN10620104 CIP         <span style="color: #555555;">&lt;=</span>0.25    <span style="color: #BB0000;">NA</span> <span style="color: #080808; background-color: #5FD7AF;">  S  </span>      <span style="color: #080808; background-color: #FF5F5F;">  R  </span> CLSI      <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> SAMN10620102 CIP         <span style="color: #555555;">&gt;=</span>4<span style="color: #555555;">.00</span>    <span style="color: #BB0000;">NA</span> <span style="color: #080808; background-color: #FF5F5F;">  R  </span>      <span style="color: #080808; background-color: #FF5F5F;">  R  </span> CLSI      <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> SAMN10620129 CIP         <span style="color: #555555;">&gt;=</span>4<span style="color: #555555;">.00</span>    <span style="color: #BB0000;">NA</span> <span style="color: #080808; background-color: #FF5F5F;">  R  </span>      <span style="color: #080808; background-color: #FF5F5F;">  R  </span> CLSI      <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> SAMN10620121 CIP         <span style="color: #555555;">&gt;=</span>4<span style="color: #555555;">.00</span>    <span style="color: #BB0000;">NA</span> <span style="color: #080808; background-color: #FF5F5F;">  R  </span>      <span style="color: #080808; background-color: #FF5F5F;">  R  </span> CLSI      <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> SAMN10620086 CIP         <span style="color: #555555;">&gt;=</span>4<span style="color: #555555;">.00</span>    <span style="color: #BB0000;">NA</span> <span style="color: #080808; background-color: #FF5F5F;">  R  </span>      <span style="color: #080808; background-color: #FF5F5F;">  R  </span> CLSI      <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 4,160 more rows</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 2 more variables: pheno_provided &lt;sir&gt;, spp_pheno &lt;mo&gt;</span></span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">ecoli_ast</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 10</span></span></span>
<span><span class="co">#&gt;   id           drug_agent     mic  disk pheno_clsi ecoff guideline method</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;ab&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;mic&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dsk&gt;</span> <span style="color: #949494; font-style: italic;">&lt;sir&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;sir&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> SAMN36015110 CIP        <span style="color: #555555;">&lt;</span>128<span style="color: #555555;">.00</span>    <span style="color: #BB0000;">NA</span> <span style="color: #080808; background-color: #FF5F5F;">  R  </span>      <span style="color: #080808; background-color: #FF5F5F;">  R  </span> CLSI      <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> SAMN11638310 CIP         256<span style="color: #555555;">.00</span>    <span style="color: #BB0000;">NA</span> <span style="color: #080808; background-color: #FF5F5F;">  R  </span>      <span style="color: #080808; background-color: #FF5F5F;">  R  </span> CLSI      <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> SAMN05729964 CIP          64<span style="color: #555555;">.00</span>    <span style="color: #BB0000;">NA</span> <span style="color: #080808; background-color: #FF5F5F;">  R  </span>      <span style="color: #080808; background-color: #FF5F5F;">  R  </span> CLSI      Etest </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> SAMN10620111 CIP         <span style="color: #555555;">&gt;=</span>4<span style="color: #555555;">.00</span>    <span style="color: #BB0000;">NA</span> <span style="color: #080808; background-color: #FF5F5F;">  R  </span>      <span style="color: #080808; background-color: #FF5F5F;">  R  </span> CLSI      <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> SAMN10620168 CIP         <span style="color: #555555;">&gt;=</span>4<span style="color: #555555;">.00</span>    <span style="color: #BB0000;">NA</span> <span style="color: #080808; background-color: #FF5F5F;">  R  </span>      <span style="color: #080808; background-color: #FF5F5F;">  R  </span> CLSI      <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> SAMN10620104 CIP         <span style="color: #555555;">&lt;=</span>0.25    <span style="color: #BB0000;">NA</span> <span style="color: #080808; background-color: #5FD7AF;">  S  </span>      <span style="color: #080808; background-color: #FF5F5F;">  R  </span> CLSI      <span style="color: #BB0000;">NA</span>    </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 2 more variables: pheno_provided &lt;sir&gt;, spp_pheno &lt;mo&gt;</span></span></span></code></pre></div>
<p>Data can be imported from various standard formats using the
<code>import_ast</code> function, and re-interpreted using latest
breakpoints and/or ECOFF. Use <code><a href="../reference/import_ast.html">?import_ast</a></code> to see the
available formats and other options.</p>
<p>If your assay data is not in a standard format, you can wrangle other
input data files into the necessary format, manually and/or with the
help of the <code>format_ast</code> function.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">?</span><span class="va">import_ast</span></span>
<span></span>
<span><span class="op">?</span><span class="va">format_ast</span></span></code></pre></div>
<p>The phenotype table is long form, with one row for each assay
measurement, i.e. one per strain/drug combination.</p>
<p>The essential columns for a phenotype table to work with
<code>AMRgen</code> functions are:</p>
<ul>
<li><p><code>id</code>: character string giving the sample name, used to
link to sample names in the genotype file (this column can have a
different name, in which case you’ll need to make sure it is the first
column in the dataframe OR pass its name to the functions using
<code>pheno_sample_col</code>)</p></li>
<li><p><code>spp_pheno</code>: species in the form of an AMR package
<code>mo</code> class (can be created from a column with species name as
string, using <code>AMR::as.mo(species_string)</code>)</p></li>
<li><p><code>drug_agent</code>: antibiotic name in the form of an AMR
package <code>ab</code> class (can be created from a column with
antibiotic name as string, using
<code>AMR::as.ab(antibiotic_string)</code>)</p></li>
<li><p>a phenotype column, e.g. the import functions output fields
<code>pheno_eucast</code>, <code>pheno_clsi</code>,
<code>pheno_provided</code>, <code>ecoff</code>: S/I/R phenotype calls
in the form of an AMR package <code>sir</code> class (can be created
from a column with phenotype values as string, using
<code>AMR::as(sir_string)</code>, or generated by interpreting MIC or
disk assay data using <code><a href="https://amr-for-r.org/reference/as.sir.html" class="external-link">AMR::as.sir</a></code>)</p></li>
</ul>
<p>If you want to do analyses with raw assay data (e.g. upset plots) you
will need that data in one or both of:</p>
<ul>
<li><p><code>mic</code>: MIC in the form of an AMR package
<code>mic</code> class (can be created from a column with assay values
as string, using <code>AMR::as.mic(mic_string)</code>)</p></li>
<li><p><code>disk</code>: disk diffusion zone diameter in the form of an
AMR package <code>disk</code> class (can be created from a column with
assay values as string, using
<code>AMR::as.disk(disk_string)</code>)</p></li>
</ul>
<p>The import functions also standardise names for the following common
fields:</p>
<ul>
<li><p><code>method</code>: The laboratory testing method (e.g., “MIC”,
“disk diffusion”, “Etest”, “agar dilution”)</p></li>
<li><p><code>platform</code>: The laboratory testing platform/instrument
if relevant (e.g., “Vitek”, “Phoenix”, “Sensititre”).</p></li>
<li><p><code>guideline</code>: The testing standard recorded in the
input file as being used to make the provided phenotype interpretations
(e.g. “CLSI”, “EUCAST”)</p></li>
<li><p><code>source</code>: An identifier for the dataset from which
each data point was sourced (e.g. study or hospital name, pubmed ID,
bioproject accession).</p></li>
</ul>
</div>
<div class="section level3">
<h3 id="plot-phenotype-data-distribution">3. Plot phenotype data distribution<a class="anchor" aria-label="anchor" href="#plot-phenotype-data-distribution"></a>
</h3>
<p>It is always a good idea to check the distribution of raw AST data
that we have to work with. The function <code><a href="../reference/assay_by_var.html">assay_by_var()</a></code> can
be used to plot the distribution of MIC or disk measurements, coloured
by a variable.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Example E. coli AST data from NCBI</span></span>
<span></span>
<span><span class="co"># Plot MIC distribution, coloured by CLSI S/I/R call</span></span>
<span><span class="fu"><a href="../reference/assay_by_var.html">assay_by_var</a></span><span class="op">(</span>pheno_table <span class="op">=</span> <span class="va">ecoli_ast</span>, antibiotic <span class="op">=</span> <span class="st">"Ciprofloxacin"</span>, measure <span class="op">=</span> <span class="st">"mic"</span>, colour_by <span class="op">=</span> <span class="st">"pheno_clsi"</span><span class="op">)</span></span></code></pre></div>
<p><img src="AnalysingGenoPhenoData_files/figure-html/plot_mic-1.png" class="r-plt" alt="" width="100%"></p>
<p>It’s a good idea to make sure that the <code>SIR</code> field in the
input data file has been interpreted correctly against the breakpoints.
The AMRgen function <code><a href="../reference/checkBreakpoints.html">checkBreakpoints()</a></code> can be used to help
look up breakpoints in the <code>AMR</code> package. Or, if you provide
the function <code><a href="../reference/assay_by_var.html">assay_by_var()</a></code> with a species and guideline,
it can look up the breakpoints and ECOFF and annotate these directly on
the plot.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Look up breakpoints recorded in the AMR package</span></span>
<span><span class="fu"><a href="../reference/checkBreakpoints.html">checkBreakpoints</a></span><span class="op">(</span>species <span class="op">=</span> <span class="st">"E. coli"</span>, guide <span class="op">=</span> <span class="st">"CLSI 2025"</span>, antibiotic <span class="op">=</span> <span class="st">"Ciprofloxacin"</span>, assay <span class="op">=</span> <span class="st">"MIC"</span><span class="op">)</span></span>
<span><span class="co">#&gt;   MIC breakpoints determined using AMR package: S &lt;= 0.25 and R &gt; 1</span></span>
<span><span class="co">#&gt; $breakpoint_S</span></span>
<span><span class="co">#&gt; [1] 0.25</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $breakpoint_R</span></span>
<span><span class="co">#&gt; [1] 1</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $bp_standard</span></span>
<span><span class="co">#&gt; [1] "-"</span></span>
<span></span>
<span><span class="co"># Specify species and guideline, to annotate with CLSI breakpoints</span></span>
<span><span class="fu"><a href="../reference/assay_by_var.html">assay_by_var</a></span><span class="op">(</span>pheno_table <span class="op">=</span> <span class="va">ecoli_ast</span>, antibiotic <span class="op">=</span> <span class="st">"Ciprofloxacin"</span>, measure <span class="op">=</span> <span class="st">"mic"</span>, colour_by <span class="op">=</span> <span class="st">"pheno_clsi"</span>, species <span class="op">=</span> <span class="st">"E. coli"</span>, guideline <span class="op">=</span> <span class="st">"CLSI 2025"</span><span class="op">)</span></span>
<span><span class="co">#&gt;   MIC breakpoints determined using AMR package: S &lt;= 0.25 and R &gt; 1</span></span></code></pre></div>
<p><img src="AnalysingGenoPhenoData_files/figure-html/plot_mic_breakpoints-1.png" class="r-plt" alt="" width="100%"></p>
<p>When aggregating AST data from different methods and sources, it is a
good idea to check the distributions broken down by method or source.
This can be done easily by passing the <code><a href="../reference/assay_by_var.html">assay_by_var()</a></code>
function a variable name to facet by, which means a separate
distribution will be plotted for each value of that variable (e.g. each
type of ‘method’ in our AST test data). Note that this public data from
NCBI includes non-standard values in the platform (Sensititre /
Sensititer) in the platform</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># specify facet_var="method" to generate facet plots by assay method</span></span>
<span><span class="va">mic_by_platform</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/assay_by_var.html">assay_by_var</a></span><span class="op">(</span>pheno_table <span class="op">=</span> <span class="va">ecoli_ast</span>, antibiotic <span class="op">=</span> <span class="st">"Ciprofloxacin"</span>, measure <span class="op">=</span> <span class="st">"mic"</span>, colour_by <span class="op">=</span> <span class="st">"pheno_clsi"</span>, species <span class="op">=</span> <span class="st">"E. coli"</span>, guideline <span class="op">=</span> <span class="st">"CLSI 2025"</span>, facet_var <span class="op">=</span> <span class="st">"method"</span><span class="op">)</span></span>
<span><span class="co">#&gt;   MIC breakpoints determined using AMR package: S &lt;= 0.25 and R &gt; 1</span></span>
<span></span>
<span><span class="va">mic_by_platform</span><span class="op">$</span><span class="va">plot</span></span>
<span><span class="co">#&gt; NULL</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="download-reference-assay-distributions-and-compare-to-your-data">4. Download reference assay distributions and compare to your
data<a class="anchor" aria-label="anchor" href="#download-reference-assay-distributions-and-compare-to-your-data"></a>
</h3>
<p>It can also be helpful to check how your MIC or disk zone
distribution compares to the reference distributions, to get a sense of
whether your assays were calibrated correctly or if there may be some
issues with a given dataset. AMRgen has functions to download the latest
reference distributions from EUCAST (mic.eucast.org), and plot them on
their own or with your data overlaid.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># get MIC distribution for ciprofloxacin, for all organisms</span></span>
<span><span class="fu"><a href="../reference/get_eucast_amr_distribution.html">get_eucast_mic_distribution</a></span><span class="op">(</span><span class="st">"cipro"</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 2,033 × 4</span></span></span>
<span><span class="co">#&gt;    microorganism              microorganism_code   mic count</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                      <span style="color: #949494; font-style: italic;">&lt;mo&gt;</span>               <span style="color: #949494; font-style: italic;">&lt;mic&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> Achromobacter xylosoxidans <span style="color: #949494;">B_</span>ACHRMB<span style="color: #949494;">_</span>XYLS      0.002     0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> Achromobacter xylosoxidans <span style="color: #949494;">B_</span>ACHRMB<span style="color: #949494;">_</span>XYLS      0.004     0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> Achromobacter xylosoxidans <span style="color: #949494;">B_</span>ACHRMB<span style="color: #949494;">_</span>XYLS      0.008     0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> Achromobacter xylosoxidans <span style="color: #949494;">B_</span>ACHRMB<span style="color: #949494;">_</span>XYLS      0.016     0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> Achromobacter xylosoxidans <span style="color: #949494;">B_</span>ACHRMB<span style="color: #949494;">_</span>XYLS      0.03<span style="color: #555555;">0</span>     0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> Achromobacter xylosoxidans <span style="color: #949494;">B_</span>ACHRMB<span style="color: #949494;">_</span>XYLS      0.06<span style="color: #555555;">0</span>     0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> Achromobacter xylosoxidans <span style="color: #949494;">B_</span>ACHRMB<span style="color: #949494;">_</span>XYLS      0.125     0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> Achromobacter xylosoxidans <span style="color: #949494;">B_</span>ACHRMB<span style="color: #949494;">_</span>XYLS      0.25<span style="color: #555555;">0</span>     1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> Achromobacter xylosoxidans <span style="color: #949494;">B_</span>ACHRMB<span style="color: #949494;">_</span>XYLS      0.5<span style="color: #555555;">00</span>     0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> Achromobacter xylosoxidans <span style="color: #949494;">B_</span>ACHRMB<span style="color: #949494;">_</span>XYLS      1<span style="color: #555555;">.000</span>     6</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 2,023 more rows</span></span></span>
<span></span>
<span><span class="co"># specify microorganism to only get results for that pathogen</span></span>
<span><span class="va">ecoli_cip_mic_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_eucast_amr_distribution.html">get_eucast_mic_distribution</a></span><span class="op">(</span><span class="st">"cipro"</span>, <span class="st">"E. coli"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># get disk diffusion data instead</span></span>
<span><span class="va">ecoli_cip_disk_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_eucast_amr_distribution.html">get_eucast_disk_distribution</a></span><span class="op">(</span><span class="st">"cipro"</span>, <span class="st">"E. coli"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># plot the MIC data</span></span>
<span><span class="va">mics</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="va">ecoli_cip_mic_data</span><span class="op">$</span><span class="va">mic</span>, <span class="va">ecoli_cip_mic_data</span><span class="op">$</span><span class="va">count</span><span class="op">)</span></span>
<span><span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/autoplot.html" class="external-link">autoplot</a></span><span class="op">(</span></span>
<span>  <span class="va">mics</span>,</span>
<span>  ab <span class="op">=</span> <span class="st">"cipro"</span>,</span>
<span>  mo <span class="op">=</span> <span class="st">"E. coli"</span>,</span>
<span>  title <span class="op">=</span> <span class="st">"E. coli cipro reference distribution"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="AnalysingGenoPhenoData_files/figure-html/get_eucast_distribution-1.png" class="r-plt" alt="" width="100%"></p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Compare reference distribution to random test data</span></span>
<span><span class="va">my_mic_values</span> <span class="op">&lt;-</span> <span class="fu">AMR</span><span class="fu">::</span><span class="fu"><a href="https://amr-for-r.org/reference/random.html" class="external-link">random_mic</a></span><span class="op">(</span><span class="fl">500</span><span class="op">)</span></span>
<span><span class="va">comparison</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_eucast_amr_distribution.html">compare_mic_with_eucast</a></span><span class="op">(</span><span class="va">my_mic_values</span>, ab <span class="op">=</span> <span class="st">"cipro"</span>, mo <span class="op">=</span> <span class="st">"E. coli"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Joining with `by = join_by(value)`</span></span>
<span><span class="va">comparison</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 25 × 3</span></span></span>
<span><span class="co">#&gt;    value     user eucast</span></span>
<span><span class="co">#&gt;  <span style="color: #BCBCBC;">*</span> <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> &lt;=0.0005    47      0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> 0.001       29      0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> 0.002       44     14</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> 0.004       33    189</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> 0.008       28   <span style="text-decoration: underline;">3</span>952</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> 0.016       34   <span style="text-decoration: underline;">7</span>238</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> 0.03         0   <span style="text-decoration: underline;">1</span>355</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> 0.032       24      0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> 0.06         0    356</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> 0.064       32      0</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 15 more rows</span></span></span>
<span><span class="co">#&gt; Use ggplot2::autoplot() on this output to visualise.</span></span>
<span><span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/autoplot.html" class="external-link">autoplot</a></span><span class="op">(</span><span class="va">comparison</span><span class="op">)</span></span></code></pre></div>
<p><img src="AnalysingGenoPhenoData_files/figure-html/compare_mic_with_eucast-1.png" class="r-plt" alt="" width="100%"></p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span></span>
<span><span class="co"># Compare reference distribution to example E. coli data</span></span>
<span><span class="va">ecoli_cip</span> <span class="op">&lt;-</span> <span class="va">ecoli_ast</span><span class="op">$</span><span class="va">mic</span><span class="op">[</span><span class="va">ecoli_ast</span><span class="op">$</span><span class="va">drug_agent</span> <span class="op">==</span> <span class="st">"CIP"</span><span class="op">]</span></span>
<span><span class="va">comparison</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_eucast_amr_distribution.html">compare_mic_with_eucast</a></span><span class="op">(</span><span class="va">ecoli_cip</span>, ab <span class="op">=</span> <span class="st">"cipro"</span>, mo <span class="op">=</span> <span class="st">"E. coli"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Joining with `by = join_by(value)`</span></span>
<span><span class="va">comparison</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 34 × 3</span></span></span>
<span><span class="co">#&gt;    value    user eucast</span></span>
<span><span class="co">#&gt;  <span style="color: #BCBCBC;">*</span> <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> 0.002       0     14</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> 0.004       0    189</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> 0.008       0   <span style="text-decoration: underline;">3</span>952</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> &lt;0.015     41      0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> &lt;=0.015  <span style="text-decoration: underline;">2</span>642      0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> 0.016       0   <span style="text-decoration: underline;">7</span>238</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> 0.03       69   <span style="text-decoration: underline;">1</span>355</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> &lt;=0.06     11      0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> 0.06        5    356</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> 0.12       34      0</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 24 more rows</span></span></span>
<span><span class="co">#&gt; Use ggplot2::autoplot() on this output to visualise.</span></span>
<span><span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/autoplot.html" class="external-link">autoplot</a></span><span class="op">(</span><span class="va">comparison</span><span class="op">)</span></span></code></pre></div>
<p><img src="AnalysingGenoPhenoData_files/figure-html/compare_mic_with_eucast-2.png" class="r-plt" alt="" width="100%"></p>
</div>
<div class="section level3">
<h3 id="combine-genotype-and-phenotype-data-for-a-given-drug">5. Combine genotype and phenotype data for a given drug<a class="anchor" aria-label="anchor" href="#combine-genotype-and-phenotype-data-for-a-given-drug"></a>
</h3>
<p>The genotype and phenotype tables can include data related to many
different drugs, but we need to analyse things one drug at a time. The
function <code><a href="../reference/get_binary_matrix.html">get_binary_matrix()</a></code> can be used to extract
phenotype data for a specified drug, and genotype data for markers
associated with a specified drug class. It returns a single dataframe
with one row per strain, for the subset of strains that appear in both
the genotype and phenotype input tables. Each row indicates, for one
strain, both the phenotypes (with SIR column, any assay columns if
desired, and boolean 1/0 coding of R and NWT status) and the genotypes
(one column per marker, with boolean 1/0 coding of marker
presence/absence).</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get matrix combining phenotype data for ciprofloxacin, binary calls for R/NWT phenotype,</span></span>
<span><span class="co">#    and genotype presence/absence data for all markers associated with the relevant drug</span></span>
<span><span class="co">#    class (which are labelled "Quinolones" in AMRFinderPlus).</span></span>
<span><span class="va">cip_bin</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_binary_matrix.html">get_binary_matrix</a></span><span class="op">(</span></span>
<span>  <span class="va">ecoli_geno</span>,</span>
<span>  <span class="va">ecoli_ast</span>,</span>
<span>  antibiotic <span class="op">=</span> <span class="st">"Ciprofloxacin"</span>,</span>
<span>  drug_class_list <span class="op">=</span> <span class="st">"Quinolones"</span>,</span>
<span>  sir_col <span class="op">=</span> <span class="st">"pheno_clsi"</span>,</span>
<span>  keep_assay_values <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  keep_assay_values_from <span class="op">=</span> <span class="st">"mic"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt;  Defining NWT in binary matrix using ecoff column provided: ecoff</span></span>
<span></span>
<span><span class="co"># check format</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">cip_bin</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 50</span></span></span>
<span><span class="co">#&gt;   id     pheno ecoff     mic     R   NWT gyrA_S83L gyrA_D87Y gyrA_D87N parC_S80I</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;sir&gt;</span> <span style="color: #949494; font-style: italic;">&lt;sir&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;mic&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> SAMN0… <span style="color: #080808; background-color: #5FD7AF;">  S  </span> <span style="color: #080808; background-color: #5FD7AF;">  S  </span> <span style="color: #555555;">&lt;=</span>0.015     0     0         0         0         0         0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> SAMN0… <span style="color: #080808; background-color: #5FD7AF;">  S  </span> <span style="color: #080808; background-color: #5FD7AF;">  S  </span> <span style="color: #555555;">&lt;=</span>0.015     0     0         0         0         0         0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> SAMN0… <span style="color: #080808; background-color: #5FD7AF;">  S  </span> <span style="color: #080808; background-color: #5FD7AF;">  S  </span> <span style="color: #555555;">&lt;=</span>0.015     0     0         0         0         0         0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> SAMN0… <span style="color: #080808; background-color: #5FD7AF;">  S  </span> <span style="color: #080808; background-color: #FF5F5F;">  R  </span>   0.25<span style="color: #555555;">0</span>     0     1         1         0         0         0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> SAMN0… <span style="color: #080808; background-color: #5FD7AF;">  S  </span> <span style="color: #080808; background-color: #FF5F5F;">  R  </span>   0.12<span style="color: #555555;">0</span>     0     1         0         1         0         0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> SAMN0… <span style="color: #080808; background-color: #5FD7AF;">  S  </span> <span style="color: #080808; background-color: #5FD7AF;">  S  </span> <span style="color: #555555;">&lt;=</span>0.015     0     0         0         0         0         0</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 40 more variables: parE_S458A &lt;dbl&gt;, parC_S80R &lt;dbl&gt;, parE_L416F &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   qnrB6 &lt;dbl&gt;, gyrA_D87G &lt;dbl&gt;, parC_S57T &lt;dbl&gt;, parC_E84A &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   soxS_A12S &lt;dbl&gt;, qnrB2 &lt;dbl&gt;, qnrS2 &lt;dbl&gt;, parC_E84K &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   parC_A56T &lt;dbl&gt;, qnrB19 &lt;dbl&gt;, `aac(6')-Ib-cr5` &lt;dbl&gt;, parC_E84V &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   parE_I529L &lt;dbl&gt;, parE_S458T &lt;dbl&gt;, parE_E460D &lt;dbl&gt;, parC_E84G &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   qnrS1 &lt;dbl&gt;, marR_S3N &lt;dbl&gt;, `aac(6')-Ib-cr` &lt;dbl&gt;, soxR_R20H &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   qnrB1 &lt;dbl&gt;, parE_I355T &lt;dbl&gt;, soxR_G121D &lt;dbl&gt;, qnrB4 &lt;dbl&gt;, qepA &lt;dbl&gt;, …</span></span></span>
<span></span>
<span><span class="co"># list colnames, to see full list of quinolone markers included</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">cip_bin</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "id"             "pheno"          "ecoff"          "mic"           </span></span>
<span><span class="co">#&gt;  [5] "R"              "NWT"            "gyrA_S83L"      "gyrA_D87Y"     </span></span>
<span><span class="co">#&gt;  [9] "gyrA_D87N"      "parC_S80I"      "parE_S458A"     "parC_S80R"     </span></span>
<span><span class="co">#&gt; [13] "parE_L416F"     "qnrB6"          "gyrA_D87G"      "parC_S57T"     </span></span>
<span><span class="co">#&gt; [17] "parC_E84A"      "soxS_A12S"      "qnrB2"          "qnrS2"         </span></span>
<span><span class="co">#&gt; [21] "parC_E84K"      "parC_A56T"      "qnrB19"         "aac(6')-Ib-cr5"</span></span>
<span><span class="co">#&gt; [25] "parC_E84V"      "parE_I529L"     "parE_S458T"     "parE_E460D"    </span></span>
<span><span class="co">#&gt; [29] "parC_E84G"      "qnrS1"          "marR_S3N"       "aac(6')-Ib-cr" </span></span>
<span><span class="co">#&gt; [33] "soxR_R20H"      "qnrB1"          "parE_I355T"     "soxR_G121D"    </span></span>
<span><span class="co">#&gt; [37] "qnrB4"          "qepA"           "gyrA_S83A"      "qnrA1"         </span></span>
<span><span class="co">#&gt; [41] "parE_D475E"     "parC_A108V"     "qepA1"          "parE_E460K"    </span></span>
<span><span class="co">#&gt; [45] "gyrA_S83W"      "marR_R77C"      "parE_L445H"     "parE_I464F"    </span></span>
<span><span class="co">#&gt; [49] "qnrB"           "acrR_R45C"</span></span></code></pre></div>
<p>This binary matrix can be used as the starting a lot of downstream
analyses.</p>
<p>For example, we can use it as input to <code>assay_by_var</code> to
plot the assay distribution coloured by presence of a particular genetic
marker</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/assay_by_var.html">assay_by_var</a></span><span class="op">(</span><span class="va">cip_bin</span>, measure <span class="op">=</span> <span class="st">"mic"</span>, colour_by <span class="op">=</span> <span class="st">"parC_S80I"</span>, antibiotic <span class="op">=</span> <span class="st">"Ciprofloxacin"</span><span class="op">)</span></span></code></pre></div>
<p><img src="AnalysingGenoPhenoData_files/figure-html/assay_by_genotype-1.png" class="r-plt" alt="" width="100%"></p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># count the number of gyrA mutations per genome</span></span>
<span><span class="va">gyrA_mut</span> <span class="op">&lt;-</span> <span class="va">cip_bin</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>gyrA_mut <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">rowSums</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/across.html" class="external-link">across</a></span><span class="op">(</span><span class="fu"><a href="https://tidyselect.r-lib.org/reference/starts_with.html" class="external-link">contains</a></span><span class="op">(</span><span class="st">"gyrA_"</span><span class="op">)</span> <span class="op">&amp;</span> <span class="fu"><a href="https://tidyselect.r-lib.org/reference/where.html" class="external-link">where</a></span><span class="op">(</span><span class="va">is.numeric</span><span class="op">)</span><span class="op">)</span>, na.rm <span class="op">=</span> <span class="cn">T</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">mic</span>, <span class="va">gyrA_mut</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># plot the MIC distribution, coloured by count of gyrA mutations</span></span>
<span><span class="va">mic_by_gyrA_count</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/assay_by_var.html">assay_by_var</a></span><span class="op">(</span><span class="va">gyrA_mut</span>, measure <span class="op">=</span> <span class="st">"mic"</span>, colour_by <span class="op">=</span> <span class="st">"gyrA_mut"</span>, colour_legend_label <span class="op">=</span> <span class="st">"No. gyrA mutations"</span>, antibiotic <span class="op">=</span> <span class="st">"Ciprofloxacin"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">mic_by_gyrA_count</span></span></code></pre></div>
<p><img src="AnalysingGenoPhenoData_files/figure-html/assay_by_genotype-2.png" class="r-plt" alt="" width="100%"></p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># count the number of genetic determinants per genome</span></span>
<span><span class="va">marker_count</span> <span class="op">&lt;-</span> <span class="va">cip_bin</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>marker_count <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">rowSums</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/across.html" class="external-link">across</a></span><span class="op">(</span><span class="fu"><a href="https://tidyselect.r-lib.org/reference/where.html" class="external-link">where</a></span><span class="op">(</span><span class="va">is.numeric</span><span class="op">)</span> <span class="op">&amp;</span> <span class="op">!</span><span class="fu"><a href="https://tidyselect.r-lib.org/reference/all_of.html" class="external-link">any_of</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"R"</span>, <span class="st">"NWT"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>, na.rm <span class="op">=</span> <span class="cn">T</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">mic</span>, <span class="va">marker_count</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># plot the MIC distribution, coloured by count of associated genetic markers</span></span>
<span><span class="va">mic_by_marker_count</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/assay_by_var.html">assay_by_var</a></span><span class="op">(</span><span class="va">marker_count</span>, measure <span class="op">=</span> <span class="st">"mic"</span>, colour_by <span class="op">=</span> <span class="st">"marker_count"</span>, colour_legend_label <span class="op">=</span> <span class="st">"No. markers detected"</span>, antibiotic <span class="op">=</span> <span class="st">"Ciprofloxacin"</span>, bar_cols <span class="op">=</span> <span class="fu">viridisLite</span><span class="fu">::</span><span class="fu"><a href="https://sjmgarnier.github.io/viridisLite/reference/viridis.html" class="external-link">viridis</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="va">marker_count</span><span class="op">$</span><span class="va">marker_count</span><span class="op">)</span> <span class="op">+</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">mic_by_marker_count</span></span></code></pre></div>
<p><img src="AnalysingGenoPhenoData_files/figure-html/assay_by_genotype-3.png" class="r-plt" alt="" width="100%"></p>
</div>
<div class="section level3">
<h3 id="model-a-binary-drug-phenotype-using-genetic-marker-presenceabsence-data">6. Model a binary drug phenotype using genetic marker
presence/absence data<a class="anchor" aria-label="anchor" href="#model-a-binary-drug-phenotype-using-genetic-marker-presenceabsence-data"></a>
</h3>
<p>Logistic regression models can be informative to get an overview of
the association between a drug resistance phenotype, and each marker
thought to be associated with the relevant drug class.</p>
<p>The <code><a href="../reference/amr_logistic.html">amr_logistic()</a></code> function uses the
<code>get_binary_matrix</code> function to generate binary-coded
genotype and phenotype data for a specified drug and class; and fits two
logistic regression models of the form
<code>R ~ marker1 + marker2 + marker3 + ...</code> and
<code>NWT ~ marker1 + marker2 + marker3 + ...</code>.</p>
<p>Note that the ‘NWT’ variable in the latter model can be taken either
from a precomputed ECOFF-based call of WT=wildtype/NWT=nonwildtype
(encoded in the input column <code>ecoff_col</code>), or computed from
the S/I/R phenotype as NWT=R/I and WT=S.</p>
<p>The <code><a href="../reference/amr_logistic.html">amr_logistic()</a></code> function can fit the model using
either the standard logistic regression approach implemented in the
<code><a href="https://rdrr.io/r/stats/glm.html" class="external-link">glm()</a></code> function, or Firth’s bias-reduced
penalized-likelihood logistic regression implemented in the
<code>logistf</code> package. The default is to use Firth’s regression,
as standard logistic regression can fail if there are too observations
in some subgroups, which happens quite often with this kind of data. To
use <code><a href="https://rdrr.io/r/stats/glm.html" class="external-link">glm()</a></code> instead, set <code>glm=TRUE</code>.</p>
<p>The function also filters out markers with too few observations in
the combined genotype/phenotype dataset. The default minimum is 10 but
this can be changed using the <code>maf</code> parameter (maf stands for
‘minor allele frequency’). If you are having trouble fitting models, it
may be because too many markers and combinations have very few
observations, and you might try increasing the <code>maf</code> value to
ensure that rare markers are excluded prior to model fitting.</p>
<p>Using this modelling approach, a negative association with a single
marker and phenotype call of R and NWT is a strong indication that
marker does not contribute to resistance. Note however that a positive
association between a marker and R or NWT does not necessarily imply the
marker is independently contributing to the resistance phenotype, as
there may be non-independence between markers that is not adequately
adjusted for by the model.</p>
<p>The function returns 4 objects:</p>
<ul>
<li><p><code>modelR, modelNWT</code>: data frames summarising each
model, with beta coefficient, lower and upper values of 95% confidence
intervals, and p-value for each marker (generated from the raw model
output using <code><a href="../reference/logistf_details.html">logistf_details()</a></code> or
<code><a href="../reference/glm_details.html">glm_details()</a></code> as relevant)</p></li>
<li><p><code>plot</code>: a ggplot2 object generated from the
<code>modelR</code> and <code>modelNWT</code> objects using the
<code><a href="../reference/compare_estimates.html">compare_estimates()</a></code> function</p></li>
<li><p><code>bin_mat</code>: the binary matrix used as input to the
regression models</p></li>
</ul>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Manually run Firth's logistic regression model using the binary matrix produced above</span></span>
<span><span class="va">dataR</span> <span class="op">&lt;-</span> <span class="va">cip_bin</span><span class="op">[</span>, <span class="fu"><a href="https://generics.r-lib.org/reference/setops.html" class="external-link">setdiff</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">cip_bin</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"id"</span>, <span class="st">"pheno"</span>, <span class="st">"ecoff"</span>, <span class="st">"mic"</span>, <span class="st">"NWT"</span><span class="op">)</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">dataR</span> <span class="op">&lt;-</span> <span class="va">dataR</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">colSums</a></span><span class="op">(</span><span class="va">dataR</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">5</span><span class="op">]</span></span>
<span><span class="va">modelR</span> <span class="op">&lt;-</span> <span class="fu">logistf</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/logistf/man/logistf.html" class="external-link">logistf</a></span><span class="op">(</span><span class="va">R</span> <span class="op">~</span> <span class="va">.</span>, data <span class="op">=</span> <span class="va">dataR</span>, pl <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in logistf::logistf(R ~ ., data = dataR, pl = FALSE): logistf.fit:</span></span>
<span><span class="co">#&gt; Maximum number of iterations for full model exceeded. Try to increase the</span></span>
<span><span class="co">#&gt; number of iterations or alter step size by passing 'logistf.control(maxit=...,</span></span>
<span><span class="co">#&gt; maxstep=...)' to parameter control</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">modelR</span><span class="op">)</span></span>
<span><span class="co">#&gt; logistf::logistf(formula = R ~ ., data = dataR, pl = FALSE)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Model fitted by Penalized ML</span></span>
<span><span class="co">#&gt; Coefficients:</span></span>
<span><span class="co">#&gt;                        coef  se(coef) lower 0.95 upper 0.95       Chisq</span></span>
<span><span class="co">#&gt; (Intercept)      -5.3126336 0.2631271 -5.8283532 -4.7969141         Inf</span></span>
<span><span class="co">#&gt; gyrA_S83L         5.0964769 0.3396242  4.4308257  5.7621280         Inf</span></span>
<span><span class="co">#&gt; gyrA_D87Y         0.6106668 1.9594688 -3.2298214  4.4511550  0.09712520</span></span>
<span><span class="co">#&gt; gyrA_D87N         1.0360455 1.2910108 -1.4942892  3.5663801  0.64401779</span></span>
<span><span class="co">#&gt; parC_S80I         3.5373730 1.2607416  1.0663649  6.0083812  7.87244336</span></span>
<span><span class="co">#&gt; parE_S458A       -0.6704877 1.4211494 -3.4558894  2.1149139  0.22258823</span></span>
<span><span class="co">#&gt; parC_S80R         0.9483222 0.9365007 -0.8871855  2.7838299  1.02540544</span></span>
<span><span class="co">#&gt; parE_L416F        1.0898410 1.4770169 -1.8050589  3.9847409  0.54444672</span></span>
<span><span class="co">#&gt; parC_S57T         1.4138484 1.4556144 -1.4391034  4.2668003  0.94343723</span></span>
<span><span class="co">#&gt; soxS_A12S         1.5835771 1.4858311 -1.3285984  4.4957527  1.13589853</span></span>
<span><span class="co">#&gt; parC_A56T         2.6649799 1.5169240 -0.3081365  5.6380963  3.08645711</span></span>
<span><span class="co">#&gt; qnrB19            5.2724510 0.4539251  4.3827740  6.1621279         Inf</span></span>
<span><span class="co">#&gt; `aac(6')-Ib-cr5`  4.2781111 1.3428296  1.6462134  6.9100089 10.14991218</span></span>
<span><span class="co">#&gt; parC_E84V        -0.5431447 1.7851855 -4.0420441  2.9557547  0.09256874</span></span>
<span><span class="co">#&gt; parE_I529L        2.0650647 0.4422569  1.1982571  2.9318724 21.80309146</span></span>
<span><span class="co">#&gt; parE_S458T       -2.7638947 1.9476308 -6.5811808  1.0533915  2.01386205</span></span>
<span><span class="co">#&gt; parE_E460D       -1.5247536 1.8858945 -5.2210388  2.1715317  0.65367903</span></span>
<span><span class="co">#&gt; parC_E84G         1.2138825 1.5879100 -1.8983640  4.3261290  0.58438832</span></span>
<span><span class="co">#&gt; qnrS1             5.5078662 0.4382199  4.6489710  6.3667613         Inf</span></span>
<span><span class="co">#&gt; marR_S3N          3.1497735 0.5128996  2.1445088  4.1550382 37.71325169</span></span>
<span><span class="co">#&gt; parE_I355T        1.9452320 0.8701522  0.2397651  3.6506990  4.99749513</span></span>
<span><span class="co">#&gt; soxR_G121D       -2.5711031 1.6085609 -5.7238245  0.5816183  2.55484162</span></span>
<span><span class="co">#&gt; qnrB4             6.9220299 1.5713224  3.8422946 10.0017653 19.40601369</span></span>
<span><span class="co">#&gt; parE_D475E       -0.7090156 1.4119171 -3.4763223  2.0582910  0.25216990</span></span>
<span><span class="co">#&gt;                             p method</span></span>
<span><span class="co">#&gt; (Intercept)      0.000000e+00      1</span></span>
<span><span class="co">#&gt; gyrA_S83L        0.000000e+00      1</span></span>
<span><span class="co">#&gt; gyrA_D87Y        7.553072e-01      1</span></span>
<span><span class="co">#&gt; gyrA_D87N        4.222596e-01      1</span></span>
<span><span class="co">#&gt; parC_S80I        5.019379e-03      1</span></span>
<span><span class="co">#&gt; parE_S458A       6.370749e-01      1</span></span>
<span><span class="co">#&gt; parC_S80R        3.112402e-01      1</span></span>
<span><span class="co">#&gt; parE_L416F       4.605957e-01      1</span></span>
<span><span class="co">#&gt; parC_S57T        3.313954e-01      1</span></span>
<span><span class="co">#&gt; soxS_A12S        2.865207e-01      1</span></span>
<span><span class="co">#&gt; parC_A56T        7.894653e-02      1</span></span>
<span><span class="co">#&gt; qnrB19           0.000000e+00      1</span></span>
<span><span class="co">#&gt; `aac(6')-Ib-cr5` 1.443081e-03      1</span></span>
<span><span class="co">#&gt; parC_E84V        7.609366e-01      1</span></span>
<span><span class="co">#&gt; parE_I529L       3.021130e-06      1</span></span>
<span><span class="co">#&gt; parE_S458T       1.558681e-01      1</span></span>
<span><span class="co">#&gt; parE_E460D       4.188004e-01      1</span></span>
<span><span class="co">#&gt; parC_E84G        4.445974e-01      1</span></span>
<span><span class="co">#&gt; qnrS1            0.000000e+00      1</span></span>
<span><span class="co">#&gt; marR_S3N         8.194598e-10      1</span></span>
<span><span class="co">#&gt; parE_I355T       2.538403e-02      1</span></span>
<span><span class="co">#&gt; soxR_G121D       1.099568e-01      1</span></span>
<span><span class="co">#&gt; qnrB4            1.056738e-05      1</span></span>
<span><span class="co">#&gt; parE_D475E       6.155513e-01      1</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Method: 1-Wald, 2-Profile penalized log-likelihood, 3-None</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Likelihood ratio test=3335.169 on 23 df, p=0, n=3630</span></span>
<span><span class="co">#&gt; Wald test = 515.415 on 23 df, p = 0</span></span>
<span></span>
<span><span class="co"># Extract model summary details using `logistf_details()`</span></span>
<span><span class="va">modelR_summary</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/logistf_details.html">logistf_details</a></span><span class="op">(</span><span class="va">modelR</span><span class="op">)</span></span>
<span></span>
<span><span class="va">modelR_summary</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 24 × 5</span></span></span>
<span><span class="co">#&gt;    marker         est ci.lower ci.upper    pval</span></span>
<span><span class="co">#&gt;  <span style="color: #BCBCBC;">*</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> (Intercept) -<span style="color: #BB0000;">5.31</span>    -<span style="color: #BB0000;">5.83</span>     -<span style="color: #BB0000;">4.80</span> 0      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> gyrA_S83L    5.10     4.43      5.76 0      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> gyrA_D87Y    0.611   -<span style="color: #BB0000;">3.23</span>      4.45 0.755  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> gyrA_D87N    1.04    -<span style="color: #BB0000;">1.49</span>      3.57 0.422  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> parC_S80I    3.54     1.07      6.01 0.005<span style="text-decoration: underline;">02</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> parE_S458A  -<span style="color: #BB0000;">0.670</span>   -<span style="color: #BB0000;">3.46</span>      2.11 0.637  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> parC_S80R    0.948   -<span style="color: #BB0000;">0.887</span>     2.78 0.311  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> parE_L416F   1.09    -<span style="color: #BB0000;">1.81</span>      3.98 0.461  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> parC_S57T    1.41    -<span style="color: #BB0000;">1.44</span>      4.27 0.331  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> soxS_A12S    1.58    -<span style="color: #BB0000;">1.33</span>      4.50 0.287  </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 14 more rows</span></span></span>
<span><span class="co">#&gt; Use ggplot2::autoplot() on this output to visualise</span></span>
<span></span>
<span><span class="co"># Plot the point estimates and 95% confidence intervals of the model</span></span>
<span><span class="fu"><a href="../reference/plot_estimates.html">plot_estimates</a></span><span class="op">(</span><span class="va">modelR_summary</span><span class="op">)</span></span></code></pre></div>
<p><img src="AnalysingGenoPhenoData_files/figure-html/manual_logistic-1.png" class="r-plt" alt="" width="100%"></p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Alternatively, use the amr_logistic() function to model R and NWT and plot the results together</span></span>
<span><span class="va">models</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/amr_logistic.html">amr_logistic</a></span><span class="op">(</span></span>
<span>  geno_table <span class="op">=</span> <span class="va">ecoli_geno</span>,</span>
<span>  pheno_table <span class="op">=</span> <span class="va">ecoli_ast</span>,</span>
<span>  sir_col <span class="op">=</span> <span class="st">"pheno_clsi"</span>,</span>
<span>  antibiotic <span class="op">=</span> <span class="st">"Ciprofloxacin"</span>,</span>
<span>  drug_class_list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Quinolones"</span><span class="op">)</span>,</span>
<span>  maf <span class="op">=</span> <span class="fl">10</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt;  Defining NWT in binary matrix using ecoff column provided: ecoff </span></span>
<span><span class="co">#&gt; ...Fitting logistic regression model to R using logistf</span></span>
<span><span class="co">#&gt;    Filtered data contains 3630 samples (793 =&gt; 1, 2837 =&gt; 0) and 19 variables.</span></span>
<span><span class="co">#&gt; Warning in logistf::logistf(R ~ ., data = to_fit, pl = FALSE): logistf.fit:</span></span>
<span><span class="co">#&gt; Maximum number of iterations for full model exceeded. Try to increase the</span></span>
<span><span class="co">#&gt; number of iterations or alter step size by passing 'logistf.control(maxit=...,</span></span>
<span><span class="co">#&gt; maxstep=...)' to parameter control</span></span>
<span><span class="co">#&gt; ...Fitting logistic regression model to NWT using logistf</span></span>
<span><span class="co">#&gt;    Filtered data contains 3630 samples (929 =&gt; 1, 2701 =&gt; 0) and 19 variables.</span></span>
<span><span class="co">#&gt; Warning in logistf::logistf(NWT ~ ., data = to_fit, pl = FALSE): logistf.fit:</span></span>
<span><span class="co">#&gt; Maximum number of iterations for full model exceeded. Try to increase the</span></span>
<span><span class="co">#&gt; number of iterations or alter step size by passing 'logistf.control(maxit=...,</span></span>
<span><span class="co">#&gt; maxstep=...)' to parameter control</span></span>
<span><span class="co">#&gt; Generating plots</span></span></code></pre></div>
<p><img src="AnalysingGenoPhenoData_files/figure-html/amr_logistic-1.png" class="r-plt" alt="" width="100%"></p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># Output tables</span></span>
<span><span class="va">models</span><span class="op">$</span><span class="va">modelR</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 20 × 5</span></span></span>
<span><span class="co">#&gt;    marker            est ci.lower ci.upper          pval</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> (Intercept)    -<span style="color: #BB0000;">5.18</span>    -<span style="color: #BB0000;">5.66</span>     -<span style="color: #BB0000;">4.70</span> 0            </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> gyrA:Ser83Leu   4.98     4.33      5.62 0            </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> gyrA:Asp87Tyr   0.917   -<span style="color: #BB0000;">2.94</span>      4.78 0.642        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> gyrA:Asp87Asn   1.19    -<span style="color: #BB0000;">1.38</span>      3.77 0.363        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> parC:Ser80Ile   3.65     1.13      6.17 0.004<span style="text-decoration: underline;">57</span>      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> parE:Ser458Ala -<span style="color: #BB0000;">1.03</span>    -<span style="color: #BB0000;">3.86</span>      1.79 0.472        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> parC:Ser80Arg   0.940   -<span style="color: #BB0000;">0.896</span>     2.78 0.316        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> parE:Leu416Phe  1.08    -<span style="color: #BB0000;">1.85</span>      4.01 0.471        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> parC:Ser57Thr   1.28    -<span style="color: #BB0000;">1.57</span>      4.13 0.377        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> soxS:Ala12Ser   1.62    -<span style="color: #BB0000;">1.31</span>      4.56 0.279        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">11</span> parC:Ala56Thr   2.54    -<span style="color: #BB0000;">0.420</span>     5.50 0.092<span style="text-decoration: underline;">6</span>       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">12</span> qnrB19          5.13     4.26      6.01 0            </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">13</span> aac(6')-Ib-cr5  4.22     1.61      6.83 0.001<span style="text-decoration: underline;">51</span>      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">14</span> parC:Glu84Val  -<span style="color: #BB0000;">0.733</span>   -<span style="color: #BB0000;">4.49</span>      3.02 0.702        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">15</span> parE:Ile529Leu  2.04     1.19      2.90 0.000<span style="text-decoration: underline;">003</span>11   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">16</span> parC:Glu84Gly   1.22    -<span style="color: #BB0000;">1.92</span>      4.35 0.448        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">17</span> qnrS1           5.38     4.54      6.22 0            </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">18</span> marR:Ser3Asn    3.04     2.06      4.02 0.000<span style="text-decoration: underline;">000</span>001<span style="text-decoration: underline;">25</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">19</span> parE:Ile355Thr  1.86     0.190     3.53 0.029<span style="text-decoration: underline;">0</span>       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">20</span> parE:Asp475Glu -<span style="color: #BB0000;">0.761</span>   -<span style="color: #BB0000;">3.48</span>      1.96 0.583</span></span>
<span><span class="co">#&gt; Use ggplot2::autoplot() on this output to visualise</span></span>
<span></span>
<span><span class="va">models</span><span class="op">$</span><span class="va">modelNWT</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 20 × 5</span></span></span>
<span><span class="co">#&gt;    marker            est ci.lower ci.upper     pval</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> (Intercept)    -<span style="color: #BB0000;">3.71</span>    -<span style="color: #BB0000;">3.97</span>    -<span style="color: #BB0000;">3.46</span>  0   <span style="color: #949494;"> </span>   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> gyrA:Ser83Leu   9.52     6.22    12.8   1.57<span style="color: #949494;">e</span><span style="color: #BB0000;">- 8</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> gyrA:Asp87Tyr   5.51     2.81     8.21  6.29<span style="color: #949494;">e</span><span style="color: #BB0000;">- 5</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> gyrA:Asp87Asn   5.14     2.08     8.21  1.00<span style="color: #949494;">e</span><span style="color: #BB0000;">- 3</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> parC:Ser80Ile  -<span style="color: #BB0000;">2.52</span>    -<span style="color: #BB0000;">6.06</span>     1.02  1.63<span style="color: #949494;">e</span><span style="color: #BB0000;">- 1</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> parE:Ser458Ala -<span style="color: #BB0000;">3.85</span>    -<span style="color: #BB0000;">7.90</span>     0.202 6.26<span style="color: #949494;">e</span><span style="color: #BB0000;">- 2</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> parC:Ser80Arg  -<span style="color: #BB0000;">3.78</span>    -<span style="color: #BB0000;">7.89</span>     0.337 7.20<span style="color: #949494;">e</span><span style="color: #BB0000;">- 2</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> parE:Leu416Phe -<span style="color: #BB0000;">4.63</span>    -<span style="color: #BB0000;">8.40</span>    -<span style="color: #BB0000;">0.858</span> 1.61<span style="color: #949494;">e</span><span style="color: #BB0000;">- 2</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> parC:Ser57Thr   0.959   -<span style="color: #BB0000;">0.709</span>    2.63  2.60<span style="color: #949494;">e</span><span style="color: #BB0000;">- 1</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> soxS:Ala12Ser   3.71     1.44     5.98  1.38<span style="color: #949494;">e</span><span style="color: #BB0000;">- 3</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">11</span> parC:Ala56Thr   1.13    -<span style="color: #BB0000;">1.78</span>     4.03  4.48<span style="color: #949494;">e</span><span style="color: #BB0000;">- 1</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">12</span> qnrB19          6.59     4.92     8.25  8.22<span style="color: #949494;">e</span><span style="color: #BB0000;">-15</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">13</span> aac(6')-Ib-cr5  5.75     2.31     9.19  1.06<span style="color: #949494;">e</span><span style="color: #BB0000;">- 3</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">14</span> parC:Glu84Val  -<span style="color: #BB0000;">4.83</span>    -<span style="color: #BB0000;">9.41</span>    -<span style="color: #BB0000;">0.251</span> 3.87<span style="color: #949494;">e</span><span style="color: #BB0000;">- 2</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">15</span> parE:Ile529Leu  2.35     1.16     3.54  1.07<span style="color: #949494;">e</span><span style="color: #BB0000;">- 4</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">16</span> parC:Glu84Gly  -<span style="color: #BB0000;">4.54</span>    -<span style="color: #BB0000;">8.41</span>    -<span style="color: #BB0000;">0.681</span> 2.11<span style="color: #949494;">e</span><span style="color: #BB0000;">- 2</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">17</span> qnrS1           6.73     5.07     8.39  1.78<span style="color: #949494;">e</span><span style="color: #BB0000;">-15</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">18</span> marR:Ser3Asn    2.91     2.22     3.59  1.11<span style="color: #949494;">e</span><span style="color: #BB0000;">-16</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">19</span> parE:Ile355Thr -<span style="color: #BB0000;">0.882</span>   -<span style="color: #BB0000;">3.65</span>     1.89  5.32<span style="color: #949494;">e</span><span style="color: #BB0000;">- 1</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">20</span> parE:Asp475Glu -<span style="color: #BB0000;">0.263</span>   -<span style="color: #BB0000;">1.64</span>     1.12  7.08<span style="color: #949494;">e</span><span style="color: #BB0000;">- 1</span></span></span>
<span><span class="co">#&gt; Use ggplot2::autoplot() on this output to visualise</span></span>
<span></span>
<span><span class="co"># Note the matrix output is the same as cip_bin, but without the MIC data as this is not required</span></span>
<span><span class="co">#    for logistic regression.</span></span>
<span><span class="va">models</span><span class="op">$</span><span class="va">bin_mat</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 3,630 × 49</span></span></span>
<span><span class="co">#&gt;    id       pheno ecoff     R   NWT gyrA..Ser83Leu gyrA..Asp87Tyr gyrA..Asp87Asn</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;sir&gt;</span> <span style="color: #949494; font-style: italic;">&lt;sir&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> SAMN031… <span style="color: #080808; background-color: #5FD7AF;">  S  </span> <span style="color: #080808; background-color: #5FD7AF;">  S  </span>     0     0              0              0              0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> SAMN031… <span style="color: #080808; background-color: #5FD7AF;">  S  </span> <span style="color: #080808; background-color: #5FD7AF;">  S  </span>     0     0              0              0              0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> SAMN031… <span style="color: #080808; background-color: #5FD7AF;">  S  </span> <span style="color: #080808; background-color: #5FD7AF;">  S  </span>     0     0              0              0              0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> SAMN031… <span style="color: #080808; background-color: #5FD7AF;">  S  </span> <span style="color: #080808; background-color: #FF5F5F;">  R  </span>     0     1              1              0              0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> SAMN031… <span style="color: #080808; background-color: #5FD7AF;">  S  </span> <span style="color: #080808; background-color: #FF5F5F;">  R  </span>     0     1              0              1              0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> SAMN031… <span style="color: #080808; background-color: #5FD7AF;">  S  </span> <span style="color: #080808; background-color: #5FD7AF;">  S  </span>     0     0              0              0              0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> SAMN031… <span style="color: #080808; background-color: #5FD7AF;">  S  </span> <span style="color: #080808; background-color: #5FD7AF;">  S  </span>     0     0              0              0              0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> SAMN031… <span style="color: #080808; background-color: #FF5F5F;">  R  </span> <span style="color: #080808; background-color: #FF5F5F;">  R  </span>     1     1              1              0              1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> SAMN031… <span style="color: #080808; background-color: #5FD7AF;">  S  </span> <span style="color: #080808; background-color: #FF5F5F;">  R  </span>     0     1              1              0              0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> SAMN031… <span style="color: #080808; background-color: #FF5F5F;">  R  </span> <span style="color: #080808; background-color: #FF5F5F;">  R  </span>     1     1              1              0              1</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 3,620 more rows</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 41 more variables: parC..Ser80Ile &lt;dbl&gt;, parE..Ser458Ala &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   parC..Ser80Arg &lt;dbl&gt;, parE..Leu416Phe &lt;dbl&gt;, qnrB6 &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   gyrA..Asp87Gly &lt;dbl&gt;, parC..Ser57Thr &lt;dbl&gt;, parC..Glu84Ala &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   soxS..Ala12Ser &lt;dbl&gt;, qnrB2 &lt;dbl&gt;, qnrS2 &lt;dbl&gt;, parC..Glu84Lys &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   parC..Ala56Thr &lt;dbl&gt;, qnrB19 &lt;dbl&gt;, `aac(6')-Ib-cr5` &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   parC..Glu84Val &lt;dbl&gt;, parE..Ile529Leu &lt;dbl&gt;, parE..Ser458Thr &lt;dbl&gt;, …</span></span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="assess-solo-positive-predictive-value-of-genetic-markers">7. Assess solo positive predictive value of genetic markers<a class="anchor" aria-label="anchor" href="#assess-solo-positive-predictive-value-of-genetic-markers"></a>
</h3>
<p>The strongest evidence of the effect of an individual genetic marker
on a drug phenotype is its positive predictive value (PPV) for
resistance amongst strains that carry this marker ‘solo’ with no other
markers known to be associated with resistance to the drug class. This
is referred to as ‘solo PPV’.</p>
<p>The function <code><a href="../reference/solo_ppv_analysis.html">solo_ppv_analysis()</a></code> takes as input our
genotype and phenotype tables, and calculates solo PPV for resistance to
a specific drug (included in our phenotype table) formarkers associated
with the specified drug class (included in our genotype table). It uses
the <code><a href="../reference/get_binary_matrix.html">get_binary_matrix()</a></code> function to first calculate the
binary matrix, then filters out all samples that have more than one
marker.</p>
<p>It then calculates for each remaining marker, amongst the genomes in
which that marker is found solo, the number of genomes, the number and
proportion that are R or NWT, and the 95% confidence intervals for these
proportions. The values are returned as a table, and also plotted so we
can easily visualise the distribution of S/I/R calls and the solo PPV
for R and NWT, for each solo marker.</p>
<p>The function returns 4 objects:</p>
<ul>
<li><p><code>solo_stats</code>: data frame containing the numbers,
proportions and confidence intervals for PPV of R and NWT
categories</p></li>
<li><p><code>amr_binary</code>: the (wide format) binary matrix for all
strains with geno/pheno data for the specified drug/class</p></li>
<li><p><code>solo_binary</code>: the (long format) binary matrix for
only those strains in which a solo marker was found, i.e. the data used
to calculate PPV</p></li>
<li><p><code>combined_plot</code>: a plot showing the distribution of
S/I/R calls and the solo PPV for R and NWT, for each solo
marker</p></li>
</ul>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Run a solo PPV analysis</span></span>
<span><span class="va">soloPPV_cipro</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/solo_ppv_analysis.html">solo_ppv_analysis</a></span><span class="op">(</span></span>
<span>  <span class="va">ecoli_geno</span>,</span>
<span>  <span class="va">ecoli_ast</span>,</span>
<span>  sir_col <span class="op">=</span> <span class="st">"pheno_clsi"</span>,</span>
<span>  antibiotic <span class="op">=</span> <span class="st">"Ciprofloxacin"</span>,</span>
<span>  drug_class_list <span class="op">=</span> <span class="st">"Quinolones"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt;  Defining NWT in binary matrix using ecoff column provided: ecoff</span></span></code></pre></div>
<p><img src="AnalysingGenoPhenoData_files/figure-html/solo_ppv_analysis-1.png" class="r-plt" alt="" width="100%"></p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># Output table</span></span>
<span><span class="va">soloPPV_cipro</span><span class="op">$</span><span class="va">solo_stats</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 36 × 8</span></span></span>
<span><span class="co">#&gt;    marker         category     x     n   ppv     se ci.lower ci.upper</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> aac(6')-Ib-cr5 R            0     1 0     0       0          0    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> gyrA_D87N      R            0     1 0     0       0          0    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> gyrA_D87Y      R            0     3 0     0       0          0    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> gyrA_S83A      R            0     3 0     0       0          0    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> gyrA_S83L      R           27    62 0.435 0.063<span style="text-decoration: underline;">0</span>  0.312      0.559</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> marR_S3N       R            4    38 0.105 0.049<span style="text-decoration: underline;">8</span>  0.007<span style="text-decoration: underline;">69</span>    0.203</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> parC_A56T      R            0     6 0     0       0          0    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> parC_S57T      R            0    23 0     0       0          0    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> parE_D475E     R            0    61 0     0       0          0    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> parE_I355T     R            0    24 0     0       0          0    </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 26 more rows</span></span></span>
<span></span>
<span><span class="co"># Interim matrices with data used to compute stats and plots</span></span>
<span><span class="va">soloPPV_cipro</span><span class="op">$</span><span class="va">solo_binary</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 306 × 9</span></span></span>
<span><span class="co">#&gt;    id           pheno ecoff   mic  disk     R   NWT marker    value</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;sir&gt;</span> <span style="color: #949494; font-style: italic;">&lt;sir&gt;</span> <span style="color: #949494; font-style: italic;">&lt;mic&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dsk&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> SAMN03177618 <span style="color: #080808; background-color: #5FD7AF;">  S  </span> <span style="color: #080808; background-color: #FF5F5F;">  R  </span>  0.25    <span style="color: #BB0000;">NA</span>     0     1 gyrA_S83L     1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> SAMN03177619 <span style="color: #080808; background-color: #5FD7AF;">  S  </span> <span style="color: #080808; background-color: #FF5F5F;">  R  </span>  0.12    <span style="color: #BB0000;">NA</span>     0     1 gyrA_D87Y     1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> SAMN03177623 <span style="color: #080808; background-color: #5FD7AF;">  S  </span> <span style="color: #080808; background-color: #FF5F5F;">  R  </span>  0.25    <span style="color: #BB0000;">NA</span>     0     1 gyrA_S83L     1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> SAMN03177631 <span style="color: #080808; background-color: #5FD7AF;">  S  </span> <span style="color: #080808; background-color: #FF5F5F;">  R  </span>  0.25    <span style="color: #BB0000;">NA</span>     0     1 gyrA_S83L     1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> SAMN03177635 <span style="color: #080808; background-color: #5FD7AF;">  S  </span> <span style="color: #080808; background-color: #FF5F5F;">  R  </span>  0.25    <span style="color: #BB0000;">NA</span>     0     1 gyrA_S83L     1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> SAMN03177637 <span style="color: #080808; background-color: #5FD7AF;">  S  </span> <span style="color: #080808; background-color: #FF5F5F;">  R  </span>  0.25    <span style="color: #BB0000;">NA</span>     0     1 gyrA_S83L     1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> SAMN03177638 <span style="color: #080808; background-color: #5FD7AF;">  S  </span> <span style="color: #080808; background-color: #FF5F5F;">  R  </span>  0.25    <span style="color: #BB0000;">NA</span>     0     1 qnrB6         1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> SAMN03177639 <span style="color: #080808; background-color: #5FD7AF;">  S  </span> <span style="color: #080808; background-color: #FF5F5F;">  R  </span>  0.12    <span style="color: #BB0000;">NA</span>     0     1 gyrA_S83L     1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> SAMN03177643 <span style="color: #080808; background-color: #5FD7AF;">  S  </span> <span style="color: #080808; background-color: #FF5F5F;">  R  </span>  0.25    <span style="color: #BB0000;">NA</span>     0     1 gyrA_S83L     1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> SAMN03177646 <span style="color: #080808; background-color: #5FD7AF;">  S  </span> <span style="color: #080808; background-color: #FF5F5F;">  R  </span>  0.25    <span style="color: #BB0000;">NA</span>     0     1 gyrA_S83L     1</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 296 more rows</span></span></span>
<span></span>
<span><span class="va">soloPPV_cipro</span><span class="op">$</span><span class="va">amr_binary</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 3,630 × 51</span></span></span>
<span><span class="co">#&gt;    id        pheno ecoff     mic  disk     R   NWT gyrA_S83L gyrA_D87Y gyrA_D87N</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;sir&gt;</span> <span style="color: #949494; font-style: italic;">&lt;sir&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;mic&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dsk&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> SAMN0317… <span style="color: #080808; background-color: #5FD7AF;">  S  </span> <span style="color: #080808; background-color: #5FD7AF;">  S  </span> <span style="color: #555555;">&lt;=</span>0.015    <span style="color: #BB0000;">NA</span>     0     0         0         0         0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> SAMN0317… <span style="color: #080808; background-color: #5FD7AF;">  S  </span> <span style="color: #080808; background-color: #5FD7AF;">  S  </span> <span style="color: #555555;">&lt;=</span>0.015    <span style="color: #BB0000;">NA</span>     0     0         0         0         0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> SAMN0317… <span style="color: #080808; background-color: #5FD7AF;">  S  </span> <span style="color: #080808; background-color: #5FD7AF;">  S  </span> <span style="color: #555555;">&lt;=</span>0.015    <span style="color: #BB0000;">NA</span>     0     0         0         0         0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> SAMN0317… <span style="color: #080808; background-color: #5FD7AF;">  S  </span> <span style="color: #080808; background-color: #FF5F5F;">  R  </span>   0.25<span style="color: #555555;">0</span>    <span style="color: #BB0000;">NA</span>     0     1         1         0         0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> SAMN0317… <span style="color: #080808; background-color: #5FD7AF;">  S  </span> <span style="color: #080808; background-color: #FF5F5F;">  R  </span>   0.12<span style="color: #555555;">0</span>    <span style="color: #BB0000;">NA</span>     0     1         0         1         0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> SAMN0317… <span style="color: #080808; background-color: #5FD7AF;">  S  </span> <span style="color: #080808; background-color: #5FD7AF;">  S  </span> <span style="color: #555555;">&lt;=</span>0.015    <span style="color: #BB0000;">NA</span>     0     0         0         0         0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> SAMN0317… <span style="color: #080808; background-color: #5FD7AF;">  S  </span> <span style="color: #080808; background-color: #5FD7AF;">  S  </span> <span style="color: #555555;">&lt;=</span>0.015    <span style="color: #BB0000;">NA</span>     0     0         0         0         0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> SAMN0317… <span style="color: #080808; background-color: #FF5F5F;">  R  </span> <span style="color: #080808; background-color: #FF5F5F;">  R  </span>  <span style="color: #555555;">&gt;</span>4<span style="color: #555555;">.000</span>    <span style="color: #BB0000;">NA</span>     1     1         1         0         1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> SAMN0317… <span style="color: #080808; background-color: #5FD7AF;">  S  </span> <span style="color: #080808; background-color: #FF5F5F;">  R  </span>   0.25<span style="color: #555555;">0</span>    <span style="color: #BB0000;">NA</span>     0     1         1         0         0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> SAMN0317… <span style="color: #080808; background-color: #FF5F5F;">  R  </span> <span style="color: #080808; background-color: #FF5F5F;">  R  </span>  <span style="color: #555555;">&gt;</span>4<span style="color: #555555;">.000</span>    <span style="color: #BB0000;">NA</span>     1     1         1         0         1</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 3,620 more rows</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 41 more variables: parC_S80I &lt;dbl&gt;, parE_S458A &lt;dbl&gt;, parC_S80R &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   parE_L416F &lt;dbl&gt;, qnrB6 &lt;dbl&gt;, gyrA_D87G &lt;dbl&gt;, parC_S57T &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   parC_E84A &lt;dbl&gt;, soxS_A12S &lt;dbl&gt;, qnrB2 &lt;dbl&gt;, qnrS2 &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   parC_E84K &lt;dbl&gt;, parC_A56T &lt;dbl&gt;, qnrB19 &lt;dbl&gt;, `aac(6')-Ib-cr5` &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   parC_E84V &lt;dbl&gt;, parE_I529L &lt;dbl&gt;, parE_S458T &lt;dbl&gt;, parE_E460D &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   parC_E84G &lt;dbl&gt;, qnrS1 &lt;dbl&gt;, marR_S3N &lt;dbl&gt;, `aac(6')-Ib-cr` &lt;dbl&gt;, …</span></span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="compare-markers-with-assay-data">8. Compare markers with assay data<a class="anchor" aria-label="anchor" href="#compare-markers-with-assay-data"></a>
</h3>
<p>So far we have considered only the impact of individual markers, and
their association with categorical S/I/R or WT/NWT calls.</p>
<p>The function <code><a href="../reference/amr_upset.html">amr_upset()</a></code> takes as binary matrix table
<code>cip_bin</code> summarising ciprofloxacin resistance vs quinolone
markers, generated using <code><a href="../reference/get_binary_matrix.html">get_binary_matrix()</a></code>, and explores
the distribution of MIC or disk diffusion assay values for all observed
combinations of markers (solo or multiple markers). It visualises the
data in the form of an upset plot, showing the distribution of assay
values and S/I/R calls for each observed marker combination, and returns
a summary of these distributions (including sample size, median and
interquartile range, number and proportion classified as R).</p>
<p>The function returns 2 objects:</p>
<ul>
<li><p><code>summary</code>: data frame containing summarising the data
associated with each combination of markers</p></li>
<li><p><code>plot</code>: an upset plot showing the distribution of
assay values, and breakdown of S/I/R calls, for each observed marker
combination</p></li>
</ul>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Compare ciprofloxacin MIC data with quinolone marker combinations,</span></span>
<span><span class="co">#    using the binary matrix we constructed earlier via get_binary_matrix()</span></span>
<span><span class="va">cipro_mic_upset</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/amr_upset.html">amr_upset</a></span><span class="op">(</span></span>
<span>  <span class="va">cip_bin</span>,</span>
<span>  min_set_size <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  assay <span class="op">=</span> <span class="st">"mic"</span>,</span>
<span>  order <span class="op">=</span> <span class="st">"value"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="AnalysingGenoPhenoData_files/figure-html/amr_upset-1.png" class="r-plt" alt="" width="100%"></p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># Output table</span></span>
<span><span class="va">cipro_mic_upset</span><span class="op">$</span><span class="va">summary</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 103 × 19</span></span></span>
<span><span class="co">#&gt;    marker_list        marker_count     n combination_id   R.n   R.ppv R.ci_lower</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> <span style="color: #949494;">""</span>                            0  <span style="text-decoration: underline;">2</span>590 0_0_0_0_0_0_0…    10 0.003<span style="text-decoration: underline;">86</span>    0.001<span style="text-decoration: underline;">47</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> <span style="color: #949494;">"</span>qnrB<span style="color: #949494;">"</span>                        1     1 0_0_0_0_0_0_0…     1 1          1      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> <span style="color: #949494;">"</span>parE_E460K, gyrA…            2     1 0_0_0_0_0_0_0…     1 1          1      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> <span style="color: #949494;">"</span>parE_D475E<span style="color: #949494;">"</span>                  1    61 0_0_0_0_0_0_0…     0 0          0      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> <span style="color: #949494;">"</span>qnrA1<span style="color: #949494;">"</span>                       1     2 0_0_0_0_0_0_0…     0 0          0      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> <span style="color: #949494;">"</span>gyrA_S83A<span style="color: #949494;">"</span>                   1     3 0_0_0_0_0_0_0…     0 0          0      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> <span style="color: #949494;">"</span>qnrB4<span style="color: #949494;">"</span>                       1     2 0_0_0_0_0_0_0…     2 1          1      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> <span style="color: #949494;">"</span>parE_I355T<span style="color: #949494;">"</span>                  1    24 0_0_0_0_0_0_0…     0 0          0      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> <span style="color: #949494;">"</span>marR_S3N<span style="color: #949494;">"</span>                    1    38 0_0_0_0_0_0_0…     4 0.105      0.007<span style="text-decoration: underline;">69</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> <span style="color: #949494;">"</span>marR_S3N, parE_D…            2     4 0_0_0_0_0_0_0…     0 0          0      </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 93 more rows</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 12 more variables: R.ci_upper &lt;dbl&gt;, NWT.n &lt;dbl&gt;, NWT.ppv &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   NWT.ci_lower &lt;dbl&gt;, NWT.ci_upper &lt;dbl&gt;, median_excludeRangeValues &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   q25_excludeRangeValues &lt;dbl&gt;, q75_excludeRangeValues &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   n_excludeRangeValues &lt;int&gt;, median_ignoreRanges &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   q25_ignoreRanges &lt;dbl&gt;, q75_ignoreRanges &lt;dbl&gt;</span></span></span></code></pre></div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Free and Open Source (GPL-3), brought by AMRverse/AMRgen Development Team</p>
</div>

<div class="pkgdown-footer-right">
  <p>Our goal is to develop interpretive standards for AMR genotypes, akin to the interpretive standards developed by EUCAST and CLSI for antimicrobial susceptibility phenotyping.</p>
</div>

    </footer>
</div>





  </body>
</html>
