# ===================================================================== #
#  Licensed as GPL-v3.0.                                                #
#                                                                       #
#  Developed as part of the AMRverse (https://github.com/AMRverse):     #
#  https://github.com/AMRverse/AMRgen                                   #
#                                                                       #
#  We created this package for both routine data analysis and academic  #
#  research and it was publicly released in the hope that it will be    #
#  useful, but it comes WITHOUT ANY WARRANTY OR LIABILITY.              #
#                                                                       #
#  This R package is free software; you can freely use and distribute   #
#  it for both personal and commercial purposes under the terms of the  #
#  GNU General Public License version 3.0 (GNU GPL-3), as published by  #
#  the Free Software Foundation.                                        #
# ===================================================================== #

#' Generate a Stacked Bar Plot of MIC Values Colored by Gene Symbol for Each Antibiotic
#'
#' This function creates a stacked bar plot using `ggplot2`, where the x-axis represents MIC (Minimum Inhibitory Concentration) values, the y-axis represents their frequency, and the bars are colored by the gene symbols. The plot is faceted by antibiotic.
#' @param pheno_data A data.frame read in by the phenotypic parser, containing a column listing the sample ID, "spp_pheno" (class mo), "drug_agent" (class ab), "mic" (class mic), "disk" (class disk). The sample ID column is used to match to geno_data.
#' @param geno_data A data.frame generated by the genotypic parser, containing a sample ID column, "marker" (class gene), "drug_agent" (class ab, note this value can be NA if there is no specific agent), and "drug_class" (controlled vocab of drug classes for the AMR package).
#' @param pathogen_mo A string in mo format specifying what pathogen is used to select breakpoints, e.g. 'B_STPHY_AURS'.
#' @param abs_to_plot A vector of strings listing the antibiotic drugs you wish to plot. Need to be in three letter code format as per the AMR package.
#' @param pheno_sample_col The name of the sample column in `pheno_data`, defaults to first column.
#' @param geno_sample_col The name of the sample column in `geno_data`, defaults to first column.
#' @param measure_method A string specifying the method used to measure antimicrobial resistance. The default is 'mic' for Minimum Inhibitory Concentration, the other option is 'disk' for Disk Diffusion.
#' @param breakpoint_source A string representing the source for breakpoint data. By default, it is set to EUCAST 2025.
#' @param breakpoint_type A string specifying the type of breakpoint to use, default is 'ECOFF'.
#' @param color_by A string specifying the variable used to color the plot elements. Options include 'gene' (default) to color by gene, or 'class' to color by drug class. Any other strings also default to gene.
#' @param plot_title A string specifying the title of the plot. The default title is "Frequency distribution of values for different genotypes".
#' @param leg_pos A string specifying the position of the plot's legend. The default is 'bottom', but other options include 'top', 'left', or 'right'.
#' @param ncol_leg An integer specifying the number of columns for the legend. By default, it is calculated based on the number of unique gene symbols in `geno_data`, rounded to 10 items per column, but minimally 2. This ensures the legend is well-organized when there are many gene symbols.
#' @details The function automatically groups the data by antibiotic, MIC (mg/L), and gene symbol and calculates the frequency of occurrences. The plot uses the `viridis` color scale with the "turbo" palette by default. You can replace it with your custom palette if desired.
#' @importFrom AMR ab_group scale_x_mic
#' @importFrom dplyr case_when filter group_by join_by left_join mutate pull select summarise
#' @importFrom ggplot2 aes element_blank element_text facet_wrap geom_bar geom_vline ggplot guide_legend guides labs theme theme_minimal unit
#' @importFrom rlang sym
#' @return A ggplot2 object representing the stacked bar plot.
#' @export
#' @examples
#' \dontrun{
#' barplot_mic_gen(pheno_data,
#'   geno_data,
#'   abs_to_plot = c("CIP", "IPM", "MEM"),
#'   measure_method = "mic",
#'   pathogen_mo = "B_ESCHR_COLI"
#' )
#' }
barplot_mic_gen <- function(pheno_data, geno_data,
                            # provides specifics or a single name if it's the same in both
                            pathogen_mo,
                            abs_to_plot, # list of drugs you want to plot, using three letter codes
                            pheno_sample_col = NULL, geno_sample_col = NULL, # default is null and assume first col in each, otherwise user
                            measure_method = "mic",
                            breakpoint_source = "EUCAST 2025",
                            breakpoint_type = "ECOFF",
                            color_by = "gene", # alt.: 'class'
                            plot_title = "Frequency distribution of values for different genotypes",
                            leg_pos = "bottom",
                            ncol_leg = min(2, round(length(unique(geno_data$marker)) / 10))) {
  # todo:
  # add a lot more parameters (ggplot2)

  # TODO (JH):
  # - fix sample ID column matching (provide as argument to function, or assume the first column and use that name)

  # 1. get breakpoint data ----
  if (toupper(breakpoint_source) %in% AMR::clinical_breakpoints$guideline) {
    breakpoints <- toupper(breakpoint_source) # Replace with actual path
    message("Using ", breakpoint_source, " breakpoints.")
  } else {
    message("Invalid breakpoint source. Using default: EUCAST 2025")
    breakpoints <- "EUCAST 2025"
  }

  if (breakpoint_type %in% AMR::clinical_breakpoints$type) {
    breakpoint_type <- breakpoint_type # Replace with actual path
  } else {
    message("Invalid breakpoint type. Using default: human")
    breakpoint_type <- "human"
  }

  if (!measure_method %in% c("mic", "disk")) {
    stop('Please choose one of the methods "mic" or "disk" for method.')
  }

  # get unique antibiotics in pheno file
  ab_bp <- data.frame("ab" = unique(pheno_data$drug_agent))

  # select only those we want to plot
  ab_bp <- ab_bp %>% filter(ab %in% abs_to_plot)

  clinical_breakpoints_here <- AMR::clinical_breakpoints %>%
    filter(
      type == breakpoint_type,
      guideline == breakpoint_source,
      method == toupper(measure_method),
      mo == pathogen_mo,
      ab %in% ab_bp$ab
    )

  ab_bp <- ab_bp %>%
    left_join(
      clinical_breakpoints_here %>%
        select(ab, breakpoint_S, breakpoint_R, disk_dose),
      by = "ab"
    )

  ## NEED TO RETURN A WARNING HERE IF THERE ARE NO GUIDELINES FOR THE CHOSEN DRUGS

  # 2. prep data ----
  # get the column names to merge on - if provided use those, otherwise use first col as default
  if (is.null(geno_sample_col)) {
    geno_sample_col <- colnames(geno_data)[1]
  }
  if (is.null(pheno_sample_col)) {
    pheno_sample_col <- colnames(pheno_data)[1]
  }
  sample_id_results <- compare_geno_pheno_id(geno_data, pheno_data, geno_sample_col, pheno_sample_col)

  # we want to select only pheno data and geno data matching our chosen Abs
  # HOWEVER - I think we want to also include the drug_class in the pheno data, because otherwise we're going to miss entries where we have no agent, but we do have the class
  ab_groups_plot <- antibiotics %>%
    filter(ab %in% abs_to_plot) %>%
    pull(group)

  pheno_data_overlap <- sample_id_results$pheno_matched %>% filter(drug_agent %in% abs_to_plot)
  geno_data_overlap <- sample_id_results$geno_matched %>% filter(drug_agent %in% abs_to_plot | drug_class %in% ab_groups_plot)

  # now we want to colour bars by combos of genes in each sample, so first need to
  # list the full set of markers for a drug in each genome
  geno_data_markers <- geno_data_overlap %>%
    group_by(Name, drug_class) %>%
    summarise(markers = paste(marker, collapse = ", "), .groups = "drop")

  # merge with the pheno data (we can now as we have one row per sample!!)
  # only merge on rows where the drug class matches
  merged_data <- pheno_data_overlap %>%
    mutate(drug_class = ab_group(drug_agent), .after = drug_agent) %>%
    left_join(geno_data_markers,
      by = c(stats::setNames(geno_sample_col, pheno_sample_col), "drug_class")
    )

  # todo add disk breakpoint
  mic_summary <- merged_data %>%
    left_join(ab_bp, join_by("drug_agent" == "ab")) %>%
    mutate(fill_col = case_when(
      color_by == "class" ~ ifelse(is.na(drug_class), "NA", drug_class),
      TRUE ~ markers
    ))

  # add in missing factor levels for the mic - want doubling dilutions
  # from the minimal value to the max value on the plot

  # 3. plot ----
  ggplot(mic_summary, aes(x = !!sym(measure_method), fill = fill_col)) +
    geom_bar() + # Use stack for stacking bars
    scale_x_mic() +
    facet_wrap(~drug_agent, scales = "free", ncol = NULL) + # Create a panel for each antibiotic
    theme_minimal() +
    # exchange to our own color palette?
    # scale_fill_viridis_d(option = "turbo") +  # Use "turbo" or other options like "viridis", "plasma"
    labs(
      title = plot_title,
      x = measure_method,
      y = "Number of isolates",
      fill = "marker"
    ) +
    theme(
      legend.position = leg_pos,
      legend.title = element_blank(),
      legend.text = element_text(size = 8), # Reduce legend text size
      legend.key.size = unit(0.4, "cm"), # Reduce the size of legend keys
      axis.text.x = element_text(hjust = 1),
      strip.text = element_text(size = 10, face = "bold") # Customize panel titles
    ) +
    guides(fill = guide_legend(ncol = ncol_leg)) + # Arrange legend items into columns
    geom_vline(
      data = ab_bp, aes(xintercept = breakpoint_S),
      color = "black", linetype = "dashed"
    ) + # Add breakpoint_S lines
    geom_vline(
      data = ab_bp, aes(xintercept = breakpoint_R),
      color = "black", linetype = "dashed"
    ) # Add breakpoint_R lines
}


#' Generate a Stacked Bar Plot of MIC Values Colored by Gene Symbol for Each Antibiotic
#'
#' This function creates a stacked bar plot using `ggplot2`, where the x-axis represents MIC (Minimum Inhibitory Concentration) or disk values, the y-axis represents their frequency, and the bars are colored to indicate whether the assay value is expressed as a range or not. It can optionally be faceted on an additional categorical variable.
#' @param pheno_table Phenotype table in standard format as per import_ast().
#' @param antibiotic Name of the antibiotic.
#' @param measure Field name containing the assay measurements to plot (default "mic").
#' @param var Field name containing a field to facet on (default "method").
#' @param species (optional) Name of species, so we can retrieve breakpoints to print at the top of the plot to help interpret it.
#' @param bp_site (optional) Breakpoint site to retrieve (only relevant if also supplying species to retrieve breakpoints).
#' @param marker_free_strains (optional) Vector of sample names to select to get their own plot. Most useful for defining the set of strains with no known markers associated with the given antibiotic, so you can view the distribution of assay values for strains expected to be wildtype, which can help to identify issues with the assay.
#' @param cols (optional) Manual colour scale to use for plot.
#' @return A list containing
#' \item{plot}{Main plot with all samples that have assay data for the given antibiotic}
#' \item{plot_nomarkers}{Additional plot showing only those samples listed in `marker_free_strains`}
#' @export
assay_by_var <- function(pheno_table, antibiotic, measure="mic", var="method", 
                         species=NULL, marker_free_strains=NULL, bp_site=NULL, 
                         cols=c(range="maroon", value="navy", `NA`="grey")) {
  if (measure %in% colnames(pheno_table)) {
    pheno_table <- pheno_table %>%
      filter(!is.na(get(measure))) %>%
      filter(drug_agent==as.ab(antibiotic)) %>%
      arrange(get(measure))
  } else {stop(paste0("No '", measure, "' column in input table"))}
  if (!(var %in% colnames(pheno_table))) {
    stop(paste0("No '", var, "' column in input table"))
  }
  
  # if species provided, check breakpoints to annotate plot
  if (!is.null(species)) {
    if (measure %in% c("mic", "disk")) {
      ecoff <- safe_execute(getBreakpoints(species=species, guide="EUCAST 2025", antibiotic=antibiotic, "ECOFF") %>% filter(method==toupper(measure)) %>% pull(breakpoint_S))
      bp_S <- safe_execute(unlist(checkBreakpoints(species=species, guide="EUCAST 2025", antibiotic=antibiotic, bp_site=bp_site, assay=toupper(measure))[1]))
      bp_R <- safe_execute(unlist(checkBreakpoints(species=species, guide="EUCAST 2025", antibiotic=antibiotic, bp_site=bp_site, assay=toupper(measure))[2]))
      if (measure=="mic") {subtitle=paste("ECOFF:", ecoff, "S <=", bp_S, "R>", bp_R)}
      else if (measure=="disk") {subtitle=paste("ECOFF:", ecoff, "S >=", bp_S, "R<", bp_R)}
    } else {subtitle=NULL}
  } else {subtitle=NULL}
  
  # plot distribution per variable
  if (nrow(pheno_table)>0) {
    plot_all <- pheno_table %>%
      mutate(range=if_else(grepl("<",get(measure)), "range", "value")) %>%
      ggplot(aes(x=factor(!!sym(measure)), fill=range)) +
      geom_bar() +
      labs(x="Measurement", y="count", fill="Value", subtitle=subtitle,
           title=paste(antibiotic, "assay distributions for all samples")) +
      theme(axis.text.x = element_text(angle = 60, vjust = 1, hjust = 1))
    if (!is.null(cols)) {
      plot_all <- plot_all + scale_fill_manual(values=cols)
    }
    if (pheno_table %>% filter(!is.na(get(var))) %>% nrow() > 0) {
      plot_all <- plot_all + facet_wrap(~get(var), ncol=1, scales="free_y")
    }
  } else {plot_all <- NULL}
  
  # samples with no markers
  if (!is.null(marker_free_strains)) {
    pheno_table_nomarkers <- pheno_table %>% filter(id %in% marker_free_strains)
    if (nrow(pheno_table_nomarkers)>0) {
      plot_nomarkers <- pheno_table_nomarkers %>%
        mutate(range=if_else(grepl("<",get(measure)), "range", "value")) %>%
        ggplot(aes(x=factor(!!sym(measure)), fill=range)) +
        geom_bar() +
        labs(x="Measurement", y="count", fill="Value", subtitle=subtitle,
             title=paste(antibiotic, "assay distributions for samples with no markers identified")) +
        theme(axis.text.x = element_text(angle = 60, vjust = 1, hjust = 1))
      if (!is.null(cols)) {
        plot_nomarkers <- plot_nomarkers + scale_fill_manual(values=cols)
      }
      if (pheno_table_nomarkers %>% filter(!is.na(get(var))) %>% nrow()>0) {
        plot_nomarkers <- plot_nomarkers + facet_wrap(~get(var), ncol=1, scales="free_y")
      }
    } else {plot_nomarkers <- NULL}
  } else {plot_nomarkers <- NULL}
  
  return(list(plot_nomarkers=plot_nomarkers, plot=plot_all))
}

